// @figma/astraui/dist/index.mjs
import { jsx as t, jsxs as p, Fragment as rr } from "react/jsx-runtime";
import { forwardRef as _e, createElement as he, useState as R, useEffect as F, useRef as X, useCallback as K, createContext as tr, useContext as or } from "react";
function Be(e) {
  var r, o, n = "";
  if (typeof e == "string" || typeof e == "number") n += e;
  else if (typeof e == "object") if (Array.isArray(e)) {
    var s = e.length;
    for (r = 0; r < s; r++) e[r] && (o = Be(e[r])) && (n && (n += " "), n += o);
  } else for (o in e) e[o] && (n && (n += " "), n += o);
  return n;
}
function nr() {
  for (var e, r, o = 0, n = "", s = arguments.length; o < s; o++) (e = arguments[o]) && (r = Be(e)) && (n && (n += " "), n += r);
  return n;
}
const sr = (e, r) => {
  const o = new Array(e.length + r.length);
  for (let n = 0; n < e.length; n++)
    o[n] = e[n];
  for (let n = 0; n < r.length; n++)
    o[e.length + n] = r[n];
  return o;
}, ar = (e, r) => ({
  classGroupId: e,
  validator: r
}), Te = (e = /* @__PURE__ */ new Map(), r = null, o) => ({
  nextPart: e,
  validators: r,
  classGroupId: o
}), ce = "-", Ne = [], ir = "arbitrary..", lr = (e) => {
  const r = dr(e), {
    conflictingClassGroups: o,
    conflictingClassGroupModifiers: n
  } = e;
  return {
    getClassGroupId: (i) => {
      if (i.startsWith("[") && i.endsWith("]"))
        return cr(i);
      const l = i.split(ce), c = l[0] === "" && l.length > 1 ? 1 : 0;
      return Ve(l, c, r);
    },
    getConflictingClassGroupIds: (i, l) => {
      if (l) {
        const c = n[i], u = o[i];
        return c ? u ? sr(u, c) : c : u || Ne;
      }
      return o[i] || Ne;
    }
  };
}, Ve = (e, r, o) => {
  if (e.length - r === 0)
    return o.classGroupId;
  const s = e[r], a = o.nextPart.get(s);
  if (a) {
    const u = Ve(e, r + 1, a);
    if (u) return u;
  }
  const i = o.validators;
  if (i === null)
    return;
  const l = r === 0 ? e.join(ce) : e.slice(r).join(ce), c = i.length;
  for (let u = 0; u < c; u++) {
    const d = i[u];
    if (d.validator(l))
      return d.classGroupId;
  }
}, cr = (e) => e.slice(1, -1).indexOf(":") === -1 ? void 0 : (() => {
  const r = e.slice(1, -1), o = r.indexOf(":"), n = r.slice(0, o);
  return n ? ir + n : void 0;
})(), dr = (e) => {
  const {
    theme: r,
    classGroups: o
  } = e;
  return ur(o, r);
}, ur = (e, r) => {
  const o = Te();
  for (const n in e) {
    const s = e[n];
    xe(s, o, n, r);
  }
  return o;
}, xe = (e, r, o, n) => {
  const s = e.length;
  for (let a = 0; a < s; a++) {
    const i = e[a];
    pr(i, r, o, n);
  }
}, pr = (e, r, o, n) => {
  if (typeof e == "string") {
    mr(e, r, o);
    return;
  }
  if (typeof e == "function") {
    fr(e, r, o, n);
    return;
  }
  hr(e, r, o, n);
}, mr = (e, r, o) => {
  const n = e === "" ? r : Pe(r, e);
  n.classGroupId = o;
}, fr = (e, r, o, n) => {
  if (br(e)) {
    xe(e(n), r, o, n);
    return;
  }
  r.validators === null && (r.validators = []), r.validators.push(ar(o, e));
}, hr = (e, r, o, n) => {
  const s = Object.entries(e), a = s.length;
  for (let i = 0; i < a; i++) {
    const [l, c] = s[i];
    xe(c, Pe(r, l), o, n);
  }
}, Pe = (e, r) => {
  let o = e;
  const n = r.split(ce), s = n.length;
  for (let a = 0; a < s; a++) {
    const i = n[a];
    let l = o.nextPart.get(i);
    l || (l = Te(), o.nextPart.set(i, l)), o = l;
  }
  return o;
}, br = (e) => "isThemeGetter" in e && e.isThemeGetter === !0, xr = (e) => {
  if (e < 1)
    return {
      get: () => {
      },
      set: () => {
      }
    };
  let r = 0, o = /* @__PURE__ */ Object.create(null), n = /* @__PURE__ */ Object.create(null);
  const s = (a, i) => {
    o[a] = i, r++, r > e && (r = 0, n = o, o = /* @__PURE__ */ Object.create(null));
  };
  return {
    get(a) {
      let i = o[a];
      if (i !== void 0)
        return i;
      if ((i = n[a]) !== void 0)
        return s(a, i), i;
    },
    set(a, i) {
      a in o ? o[a] = i : s(a, i);
    }
  };
}, be = "!", Le = ":", gr = [], ze = (e, r, o, n, s) => ({
  modifiers: e,
  hasImportantModifier: r,
  baseClassName: o,
  maybePostfixModifierPosition: n,
  isExternal: s
}), vr = (e) => {
  const {
    prefix: r,
    experimentalParseClassName: o
  } = e;
  let n = (s) => {
    const a = [];
    let i = 0, l = 0, c = 0, u;
    const d = s.length;
    for (let C = 0; C < d; C++) {
      const k = s[C];
      if (i === 0 && l === 0) {
        if (k === Le) {
          a.push(s.slice(c, C)), c = C + 1;
          continue;
        }
        if (k === "/") {
          u = C;
          continue;
        }
      }
      k === "[" ? i++ : k === "]" ? i-- : k === "(" ? l++ : k === ")" && l--;
    }
    const b = a.length === 0 ? s : s.slice(c);
    let w = b, v = !1;
    b.endsWith(be) ? (w = b.slice(0, -1), v = !0) : (
      /**
       * In Tailwind CSS v3 the important modifier was at the start of the base class name. This is still supported for legacy reasons.
       * @see https://github.com/dcastil/tailwind-merge/issues/513#issuecomment-2614029864
       */
      b.startsWith(be) && (w = b.slice(1), v = !0)
    );
    const M = u && u > c ? u - c : void 0;
    return ze(a, v, w, M);
  };
  if (r) {
    const s = r + Le, a = n;
    n = (i) => i.startsWith(s) ? a(i.slice(s.length)) : ze(gr, !1, i, void 0, !0);
  }
  if (o) {
    const s = n;
    n = (a) => o({
      className: a,
      parseClassName: s
    });
  }
  return n;
}, yr = (e) => {
  const r = /* @__PURE__ */ new Map();
  return e.orderSensitiveModifiers.forEach((o, n) => {
    r.set(o, 1e6 + n);
  }), (o) => {
    const n = [];
    let s = [];
    for (let a = 0; a < o.length; a++) {
      const i = o[a], l = i[0] === "[", c = r.has(i);
      l || c ? (s.length > 0 && (s.sort(), n.push(...s), s = []), n.push(i)) : s.push(i);
    }
    return s.length > 0 && (s.sort(), n.push(...s)), n;
  };
}, wr = (e) => ({
  cache: xr(e.cacheSize),
  parseClassName: vr(e),
  sortModifiers: yr(e),
  ...lr(e)
}), kr = /\s+/, Cr = (e, r) => {
  const {
    parseClassName: o,
    getClassGroupId: n,
    getConflictingClassGroupIds: s,
    sortModifiers: a
  } = r, i = [], l = e.trim().split(kr);
  let c = "";
  for (let u = l.length - 1; u >= 0; u -= 1) {
    const d = l[u], {
      isExternal: b,
      modifiers: w,
      hasImportantModifier: v,
      baseClassName: M,
      maybePostfixModifierPosition: C
    } = o(d);
    if (b) {
      c = d + (c.length > 0 ? " " + c : c);
      continue;
    }
    let k = !!C, A = n(k ? M.substring(0, C) : M);
    if (!A) {
      if (!k) {
        c = d + (c.length > 0 ? " " + c : c);
        continue;
      }
      if (A = n(M), !A) {
        c = d + (c.length > 0 ? " " + c : c);
        continue;
      }
      k = !1;
    }
    const j = w.length === 0 ? "" : w.length === 1 ? w[0] : a(w).join(":"), T = v ? j + be : j, N = T + A;
    if (i.indexOf(N) > -1)
      continue;
    i.push(N);
    const I = s(A, k);
    for (let V = 0; V < I.length; ++V) {
      const G = I[V];
      i.push(T + G);
    }
    c = d + (c.length > 0 ? " " + c : c);
  }
  return c;
}, Nr = (...e) => {
  let r = 0, o, n, s = "";
  for (; r < e.length; )
    (o = e[r++]) && (n = Ee(o)) && (s && (s += " "), s += n);
  return s;
}, Ee = (e) => {
  if (typeof e == "string")
    return e;
  let r, o = "";
  for (let n = 0; n < e.length; n++)
    e[n] && (r = Ee(e[n])) && (o && (o += " "), o += r);
  return o;
}, Lr = (e, ...r) => {
  let o, n, s, a;
  const i = (c) => {
    const u = r.reduce((d, b) => b(d), e());
    return o = wr(u), n = o.cache.get, s = o.cache.set, a = l, l(c);
  }, l = (c) => {
    const u = n(c);
    if (u)
      return u;
    const d = Cr(c, o);
    return s(c, d), d;
  };
  return a = i, (...c) => a(Nr(...c));
}, zr = [], S = (e) => {
  const r = (o) => o[e] || zr;
  return r.isThemeGetter = !0, r;
}, Ge = /^\[(?:(\w[\w-]*):)?(.+)\]$/i, Oe = /^\((?:(\w[\w-]*):)?(.+)\)$/i, Mr = /^\d+(?:\.\d+)?\/\d+(?:\.\d+)?$/, Ar = /^(\d+(\.\d+)?)?(xs|sm|md|lg|xl)$/, jr = /\d+(%|px|r?em|[sdl]?v([hwib]|min|max)|pt|pc|in|cm|mm|cap|ch|ex|r?lh|cq(w|h|i|b|min|max))|\b(calc|min|max|clamp)\(.+\)|^0$/, Ir = /^(rgba?|hsla?|hwb|(ok)?(lab|lch)|color-mix)\(.+\)$/, Sr = /^(inset_)?-?((\d+)?\.?(\d+)[a-z]+|0)_-?((\d+)?\.?(\d+)[a-z]+|0)/, Rr = /^(url|image|image-set|cross-fade|element|(repeating-)?(linear|radial|conic)-gradient)\(.+\)$/, H = (e) => Mr.test(e), y = (e) => !!e && !Number.isNaN(Number(e)), U = (e) => !!e && Number.isInteger(Number(e)), fe = (e) => e.endsWith("%") && y(e.slice(0, -1)), D = (e) => Ar.test(e), Fe = () => !0, _r = (e) => (
  // `colorFunctionRegex` check is necessary because color functions can have percentages in them which which would be incorrectly classified as lengths.
  // For example, `hsl(0 0% 0%)` would be classified as a length without this check.
  // I could also use lookbehind assertion in `lengthUnitRegex` but that isn't supported widely enough.
  jr.test(e) && !Ir.test(e)
), ge = () => !1, Br = (e) => Sr.test(e), Tr = (e) => Rr.test(e), Vr = (e) => !m(e) && !f(e), Pr = (e) => Z(e, $e, ge), m = (e) => Ge.test(e), q = (e) => Z(e, He, _r), Me = (e) => Z(e, Hr, y), Er = (e) => Z(e, Ze, Fe), Gr = (e) => Z(e, Ue, ge), Ae = (e) => Z(e, We, ge), Or = (e) => Z(e, De, Tr), ae = (e) => Z(e, Ye, Br), f = (e) => Oe.test(e), re = (e) => J(e, He), Fr = (e) => J(e, Ue), je = (e) => J(e, We), Wr = (e) => J(e, $e), Dr = (e) => J(e, De), ie = (e) => J(e, Ye, !0), $r = (e) => J(e, Ze, !0), Z = (e, r, o) => {
  const n = Ge.exec(e);
  return n ? n[1] ? r(n[1]) : o(n[2]) : !1;
}, J = (e, r, o = !1) => {
  const n = Oe.exec(e);
  return n ? n[1] ? r(n[1]) : o : !1;
}, We = (e) => e === "position" || e === "percentage", De = (e) => e === "image" || e === "url", $e = (e) => e === "length" || e === "size" || e === "bg-size", He = (e) => e === "length", Hr = (e) => e === "number", Ue = (e) => e === "family-name", Ze = (e) => e === "number" || e === "weight", Ye = (e) => e === "shadow", Ur = () => {
  const e = S("color"), r = S("font"), o = S("text"), n = S("font-weight"), s = S("tracking"), a = S("leading"), i = S("breakpoint"), l = S("container"), c = S("spacing"), u = S("radius"), d = S("shadow"), b = S("inset-shadow"), w = S("text-shadow"), v = S("drop-shadow"), M = S("blur"), C = S("perspective"), k = S("aspect"), A = S("ease"), j = S("animate"), T = () => ["auto", "avoid", "all", "avoid-page", "page", "left", "right", "column"], N = () => [
    "center",
    "top",
    "bottom",
    "left",
    "right",
    "top-left",
    // Deprecated since Tailwind CSS v4.1.0, see https://github.com/tailwindlabs/tailwindcss/pull/17378
    "left-top",
    "top-right",
    // Deprecated since Tailwind CSS v4.1.0, see https://github.com/tailwindlabs/tailwindcss/pull/17378
    "right-top",
    "bottom-right",
    // Deprecated since Tailwind CSS v4.1.0, see https://github.com/tailwindlabs/tailwindcss/pull/17378
    "right-bottom",
    "bottom-left",
    // Deprecated since Tailwind CSS v4.1.0, see https://github.com/tailwindlabs/tailwindcss/pull/17378
    "left-bottom"
  ], I = () => [...N(), f, m], V = () => ["auto", "hidden", "clip", "visible", "scroll"], G = () => ["auto", "contain", "none"], x = () => [f, m, c], _ = () => [H, "full", "auto", ...x()], ee = () => [U, "none", "subgrid", f, m], L = () => ["auto", {
    span: ["full", U, f, m]
  }, U, f, m], z = () => [U, "auto", f, m], $ = () => ["auto", "min", "max", "fr", f, m], W = () => ["start", "end", "center", "between", "around", "evenly", "stretch", "baseline", "center-safe", "end-safe"], Q = () => ["start", "end", "center", "stretch", "center-safe", "end-safe"], O = () => ["auto", ...x()], Y = () => [H, "auto", "full", "dvw", "dvh", "lvw", "lvh", "svw", "svh", "min", "max", "fit", ...x()], de = () => [H, "screen", "full", "dvw", "lvw", "svw", "min", "max", "fit", ...x()], ue = () => [H, "screen", "full", "lh", "dvh", "lvh", "svh", "min", "max", "fit", ...x()], g = () => [e, f, m], ve = () => [...N(), je, Ae, {
    position: [f, m]
  }], ye = () => ["no-repeat", {
    repeat: ["", "x", "y", "space", "round"]
  }], we = () => ["auto", "cover", "contain", Wr, Pr, {
    size: [f, m]
  }], pe = () => [fe, re, q], P = () => [
    // Deprecated since Tailwind CSS v4.0.0
    "",
    "none",
    "full",
    u,
    f,
    m
  ], E = () => ["", y, re, q], te = () => ["solid", "dashed", "dotted", "double"], ke = () => ["normal", "multiply", "screen", "overlay", "darken", "lighten", "color-dodge", "color-burn", "hard-light", "soft-light", "difference", "exclusion", "hue", "saturation", "color", "luminosity"], B = () => [y, fe, je, Ae], Ce = () => [
    // Deprecated since Tailwind CSS v4.0.0
    "",
    "none",
    M,
    f,
    m
  ], oe = () => ["none", y, f, m], ne = () => ["none", y, f, m], me = () => [y, f, m], se = () => [H, "full", ...x()];
  return {
    cacheSize: 500,
    theme: {
      animate: ["spin", "ping", "pulse", "bounce"],
      aspect: ["video"],
      blur: [D],
      breakpoint: [D],
      color: [Fe],
      container: [D],
      "drop-shadow": [D],
      ease: ["in", "out", "in-out"],
      font: [Vr],
      "font-weight": ["thin", "extralight", "light", "normal", "medium", "semibold", "bold", "extrabold", "black"],
      "inset-shadow": [D],
      leading: ["none", "tight", "snug", "normal", "relaxed", "loose"],
      perspective: ["dramatic", "near", "normal", "midrange", "distant", "none"],
      radius: [D],
      shadow: [D],
      spacing: ["px", y],
      text: [D],
      "text-shadow": [D],
      tracking: ["tighter", "tight", "normal", "wide", "wider", "widest"]
    },
    classGroups: {
      // --------------
      // --- Layout ---
      // --------------
      /**
       * Aspect Ratio
       * @see https://tailwindcss.com/docs/aspect-ratio
       */
      aspect: [{
        aspect: ["auto", "square", H, m, f, k]
      }],
      /**
       * Container
       * @see https://tailwindcss.com/docs/container
       * @deprecated since Tailwind CSS v4.0.0
       */
      container: ["container"],
      /**
       * Columns
       * @see https://tailwindcss.com/docs/columns
       */
      columns: [{
        columns: [y, m, f, l]
      }],
      /**
       * Break After
       * @see https://tailwindcss.com/docs/break-after
       */
      "break-after": [{
        "break-after": T()
      }],
      /**
       * Break Before
       * @see https://tailwindcss.com/docs/break-before
       */
      "break-before": [{
        "break-before": T()
      }],
      /**
       * Break Inside
       * @see https://tailwindcss.com/docs/break-inside
       */
      "break-inside": [{
        "break-inside": ["auto", "avoid", "avoid-page", "avoid-column"]
      }],
      /**
       * Box Decoration Break
       * @see https://tailwindcss.com/docs/box-decoration-break
       */
      "box-decoration": [{
        "box-decoration": ["slice", "clone"]
      }],
      /**
       * Box Sizing
       * @see https://tailwindcss.com/docs/box-sizing
       */
      box: [{
        box: ["border", "content"]
      }],
      /**
       * Display
       * @see https://tailwindcss.com/docs/display
       */
      display: ["block", "inline-block", "inline", "flex", "inline-flex", "table", "inline-table", "table-caption", "table-cell", "table-column", "table-column-group", "table-footer-group", "table-header-group", "table-row-group", "table-row", "flow-root", "grid", "inline-grid", "contents", "list-item", "hidden"],
      /**
       * Screen Reader Only
       * @see https://tailwindcss.com/docs/display#screen-reader-only
       */
      sr: ["sr-only", "not-sr-only"],
      /**
       * Floats
       * @see https://tailwindcss.com/docs/float
       */
      float: [{
        float: ["right", "left", "none", "start", "end"]
      }],
      /**
       * Clear
       * @see https://tailwindcss.com/docs/clear
       */
      clear: [{
        clear: ["left", "right", "both", "none", "start", "end"]
      }],
      /**
       * Isolation
       * @see https://tailwindcss.com/docs/isolation
       */
      isolation: ["isolate", "isolation-auto"],
      /**
       * Object Fit
       * @see https://tailwindcss.com/docs/object-fit
       */
      "object-fit": [{
        object: ["contain", "cover", "fill", "none", "scale-down"]
      }],
      /**
       * Object Position
       * @see https://tailwindcss.com/docs/object-position
       */
      "object-position": [{
        object: I()
      }],
      /**
       * Overflow
       * @see https://tailwindcss.com/docs/overflow
       */
      overflow: [{
        overflow: V()
      }],
      /**
       * Overflow X
       * @see https://tailwindcss.com/docs/overflow
       */
      "overflow-x": [{
        "overflow-x": V()
      }],
      /**
       * Overflow Y
       * @see https://tailwindcss.com/docs/overflow
       */
      "overflow-y": [{
        "overflow-y": V()
      }],
      /**
       * Overscroll Behavior
       * @see https://tailwindcss.com/docs/overscroll-behavior
       */
      overscroll: [{
        overscroll: G()
      }],
      /**
       * Overscroll Behavior X
       * @see https://tailwindcss.com/docs/overscroll-behavior
       */
      "overscroll-x": [{
        "overscroll-x": G()
      }],
      /**
       * Overscroll Behavior Y
       * @see https://tailwindcss.com/docs/overscroll-behavior
       */
      "overscroll-y": [{
        "overscroll-y": G()
      }],
      /**
       * Position
       * @see https://tailwindcss.com/docs/position
       */
      position: ["static", "fixed", "absolute", "relative", "sticky"],
      /**
       * Inset
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      inset: [{
        inset: _()
      }],
      /**
       * Inset Inline
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      "inset-x": [{
        "inset-x": _()
      }],
      /**
       * Inset Block
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      "inset-y": [{
        "inset-y": _()
      }],
      /**
       * Inset Inline Start
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       * @todo class group will be renamed to `inset-s` in next major release
       */
      start: [{
        "inset-s": _(),
        /**
         * @deprecated since Tailwind CSS v4.2.0 in favor of `inset-s-*` utilities.
         * @see https://github.com/tailwindlabs/tailwindcss/pull/19613
         */
        start: _()
      }],
      /**
       * Inset Inline End
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       * @todo class group will be renamed to `inset-e` in next major release
       */
      end: [{
        "inset-e": _(),
        /**
         * @deprecated since Tailwind CSS v4.2.0 in favor of `inset-e-*` utilities.
         * @see https://github.com/tailwindlabs/tailwindcss/pull/19613
         */
        end: _()
      }],
      /**
       * Inset Block Start
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      "inset-bs": [{
        "inset-bs": _()
      }],
      /**
       * Inset Block End
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      "inset-be": [{
        "inset-be": _()
      }],
      /**
       * Top
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      top: [{
        top: _()
      }],
      /**
       * Right
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      right: [{
        right: _()
      }],
      /**
       * Bottom
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      bottom: [{
        bottom: _()
      }],
      /**
       * Left
       * @see https://tailwindcss.com/docs/top-right-bottom-left
       */
      left: [{
        left: _()
      }],
      /**
       * Visibility
       * @see https://tailwindcss.com/docs/visibility
       */
      visibility: ["visible", "invisible", "collapse"],
      /**
       * Z-Index
       * @see https://tailwindcss.com/docs/z-index
       */
      z: [{
        z: [U, "auto", f, m]
      }],
      // ------------------------
      // --- Flexbox and Grid ---
      // ------------------------
      /**
       * Flex Basis
       * @see https://tailwindcss.com/docs/flex-basis
       */
      basis: [{
        basis: [H, "full", "auto", l, ...x()]
      }],
      /**
       * Flex Direction
       * @see https://tailwindcss.com/docs/flex-direction
       */
      "flex-direction": [{
        flex: ["row", "row-reverse", "col", "col-reverse"]
      }],
      /**
       * Flex Wrap
       * @see https://tailwindcss.com/docs/flex-wrap
       */
      "flex-wrap": [{
        flex: ["nowrap", "wrap", "wrap-reverse"]
      }],
      /**
       * Flex
       * @see https://tailwindcss.com/docs/flex
       */
      flex: [{
        flex: [y, H, "auto", "initial", "none", m]
      }],
      /**
       * Flex Grow
       * @see https://tailwindcss.com/docs/flex-grow
       */
      grow: [{
        grow: ["", y, f, m]
      }],
      /**
       * Flex Shrink
       * @see https://tailwindcss.com/docs/flex-shrink
       */
      shrink: [{
        shrink: ["", y, f, m]
      }],
      /**
       * Order
       * @see https://tailwindcss.com/docs/order
       */
      order: [{
        order: [U, "first", "last", "none", f, m]
      }],
      /**
       * Grid Template Columns
       * @see https://tailwindcss.com/docs/grid-template-columns
       */
      "grid-cols": [{
        "grid-cols": ee()
      }],
      /**
       * Grid Column Start / End
       * @see https://tailwindcss.com/docs/grid-column
       */
      "col-start-end": [{
        col: L()
      }],
      /**
       * Grid Column Start
       * @see https://tailwindcss.com/docs/grid-column
       */
      "col-start": [{
        "col-start": z()
      }],
      /**
       * Grid Column End
       * @see https://tailwindcss.com/docs/grid-column
       */
      "col-end": [{
        "col-end": z()
      }],
      /**
       * Grid Template Rows
       * @see https://tailwindcss.com/docs/grid-template-rows
       */
      "grid-rows": [{
        "grid-rows": ee()
      }],
      /**
       * Grid Row Start / End
       * @see https://tailwindcss.com/docs/grid-row
       */
      "row-start-end": [{
        row: L()
      }],
      /**
       * Grid Row Start
       * @see https://tailwindcss.com/docs/grid-row
       */
      "row-start": [{
        "row-start": z()
      }],
      /**
       * Grid Row End
       * @see https://tailwindcss.com/docs/grid-row
       */
      "row-end": [{
        "row-end": z()
      }],
      /**
       * Grid Auto Flow
       * @see https://tailwindcss.com/docs/grid-auto-flow
       */
      "grid-flow": [{
        "grid-flow": ["row", "col", "dense", "row-dense", "col-dense"]
      }],
      /**
       * Grid Auto Columns
       * @see https://tailwindcss.com/docs/grid-auto-columns
       */
      "auto-cols": [{
        "auto-cols": $()
      }],
      /**
       * Grid Auto Rows
       * @see https://tailwindcss.com/docs/grid-auto-rows
       */
      "auto-rows": [{
        "auto-rows": $()
      }],
      /**
       * Gap
       * @see https://tailwindcss.com/docs/gap
       */
      gap: [{
        gap: x()
      }],
      /**
       * Gap X
       * @see https://tailwindcss.com/docs/gap
       */
      "gap-x": [{
        "gap-x": x()
      }],
      /**
       * Gap Y
       * @see https://tailwindcss.com/docs/gap
       */
      "gap-y": [{
        "gap-y": x()
      }],
      /**
       * Justify Content
       * @see https://tailwindcss.com/docs/justify-content
       */
      "justify-content": [{
        justify: [...W(), "normal"]
      }],
      /**
       * Justify Items
       * @see https://tailwindcss.com/docs/justify-items
       */
      "justify-items": [{
        "justify-items": [...Q(), "normal"]
      }],
      /**
       * Justify Self
       * @see https://tailwindcss.com/docs/justify-self
       */
      "justify-self": [{
        "justify-self": ["auto", ...Q()]
      }],
      /**
       * Align Content
       * @see https://tailwindcss.com/docs/align-content
       */
      "align-content": [{
        content: ["normal", ...W()]
      }],
      /**
       * Align Items
       * @see https://tailwindcss.com/docs/align-items
       */
      "align-items": [{
        items: [...Q(), {
          baseline: ["", "last"]
        }]
      }],
      /**
       * Align Self
       * @see https://tailwindcss.com/docs/align-self
       */
      "align-self": [{
        self: ["auto", ...Q(), {
          baseline: ["", "last"]
        }]
      }],
      /**
       * Place Content
       * @see https://tailwindcss.com/docs/place-content
       */
      "place-content": [{
        "place-content": W()
      }],
      /**
       * Place Items
       * @see https://tailwindcss.com/docs/place-items
       */
      "place-items": [{
        "place-items": [...Q(), "baseline"]
      }],
      /**
       * Place Self
       * @see https://tailwindcss.com/docs/place-self
       */
      "place-self": [{
        "place-self": ["auto", ...Q()]
      }],
      // Spacing
      /**
       * Padding
       * @see https://tailwindcss.com/docs/padding
       */
      p: [{
        p: x()
      }],
      /**
       * Padding Inline
       * @see https://tailwindcss.com/docs/padding
       */
      px: [{
        px: x()
      }],
      /**
       * Padding Block
       * @see https://tailwindcss.com/docs/padding
       */
      py: [{
        py: x()
      }],
      /**
       * Padding Inline Start
       * @see https://tailwindcss.com/docs/padding
       */
      ps: [{
        ps: x()
      }],
      /**
       * Padding Inline End
       * @see https://tailwindcss.com/docs/padding
       */
      pe: [{
        pe: x()
      }],
      /**
       * Padding Block Start
       * @see https://tailwindcss.com/docs/padding
       */
      pbs: [{
        pbs: x()
      }],
      /**
       * Padding Block End
       * @see https://tailwindcss.com/docs/padding
       */
      pbe: [{
        pbe: x()
      }],
      /**
       * Padding Top
       * @see https://tailwindcss.com/docs/padding
       */
      pt: [{
        pt: x()
      }],
      /**
       * Padding Right
       * @see https://tailwindcss.com/docs/padding
       */
      pr: [{
        pr: x()
      }],
      /**
       * Padding Bottom
       * @see https://tailwindcss.com/docs/padding
       */
      pb: [{
        pb: x()
      }],
      /**
       * Padding Left
       * @see https://tailwindcss.com/docs/padding
       */
      pl: [{
        pl: x()
      }],
      /**
       * Margin
       * @see https://tailwindcss.com/docs/margin
       */
      m: [{
        m: O()
      }],
      /**
       * Margin Inline
       * @see https://tailwindcss.com/docs/margin
       */
      mx: [{
        mx: O()
      }],
      /**
       * Margin Block
       * @see https://tailwindcss.com/docs/margin
       */
      my: [{
        my: O()
      }],
      /**
       * Margin Inline Start
       * @see https://tailwindcss.com/docs/margin
       */
      ms: [{
        ms: O()
      }],
      /**
       * Margin Inline End
       * @see https://tailwindcss.com/docs/margin
       */
      me: [{
        me: O()
      }],
      /**
       * Margin Block Start
       * @see https://tailwindcss.com/docs/margin
       */
      mbs: [{
        mbs: O()
      }],
      /**
       * Margin Block End
       * @see https://tailwindcss.com/docs/margin
       */
      mbe: [{
        mbe: O()
      }],
      /**
       * Margin Top
       * @see https://tailwindcss.com/docs/margin
       */
      mt: [{
        mt: O()
      }],
      /**
       * Margin Right
       * @see https://tailwindcss.com/docs/margin
       */
      mr: [{
        mr: O()
      }],
      /**
       * Margin Bottom
       * @see https://tailwindcss.com/docs/margin
       */
      mb: [{
        mb: O()
      }],
      /**
       * Margin Left
       * @see https://tailwindcss.com/docs/margin
       */
      ml: [{
        ml: O()
      }],
      /**
       * Space Between X
       * @see https://tailwindcss.com/docs/margin#adding-space-between-children
       */
      "space-x": [{
        "space-x": x()
      }],
      /**
       * Space Between X Reverse
       * @see https://tailwindcss.com/docs/margin#adding-space-between-children
       */
      "space-x-reverse": ["space-x-reverse"],
      /**
       * Space Between Y
       * @see https://tailwindcss.com/docs/margin#adding-space-between-children
       */
      "space-y": [{
        "space-y": x()
      }],
      /**
       * Space Between Y Reverse
       * @see https://tailwindcss.com/docs/margin#adding-space-between-children
       */
      "space-y-reverse": ["space-y-reverse"],
      // --------------
      // --- Sizing ---
      // --------------
      /**
       * Size
       * @see https://tailwindcss.com/docs/width#setting-both-width-and-height
       */
      size: [{
        size: Y()
      }],
      /**
       * Inline Size
       * @see https://tailwindcss.com/docs/width
       */
      "inline-size": [{
        inline: ["auto", ...de()]
      }],
      /**
       * Min-Inline Size
       * @see https://tailwindcss.com/docs/min-width
       */
      "min-inline-size": [{
        "min-inline": ["auto", ...de()]
      }],
      /**
       * Max-Inline Size
       * @see https://tailwindcss.com/docs/max-width
       */
      "max-inline-size": [{
        "max-inline": ["none", ...de()]
      }],
      /**
       * Block Size
       * @see https://tailwindcss.com/docs/height
       */
      "block-size": [{
        block: ["auto", ...ue()]
      }],
      /**
       * Min-Block Size
       * @see https://tailwindcss.com/docs/min-height
       */
      "min-block-size": [{
        "min-block": ["auto", ...ue()]
      }],
      /**
       * Max-Block Size
       * @see https://tailwindcss.com/docs/max-height
       */
      "max-block-size": [{
        "max-block": ["none", ...ue()]
      }],
      /**
       * Width
       * @see https://tailwindcss.com/docs/width
       */
      w: [{
        w: [l, "screen", ...Y()]
      }],
      /**
       * Min-Width
       * @see https://tailwindcss.com/docs/min-width
       */
      "min-w": [{
        "min-w": [
          l,
          "screen",
          /** Deprecated. @see https://github.com/tailwindlabs/tailwindcss.com/issues/2027#issuecomment-2620152757 */
          "none",
          ...Y()
        ]
      }],
      /**
       * Max-Width
       * @see https://tailwindcss.com/docs/max-width
       */
      "max-w": [{
        "max-w": [
          l,
          "screen",
          "none",
          /** Deprecated since Tailwind CSS v4.0.0. @see https://github.com/tailwindlabs/tailwindcss.com/issues/2027#issuecomment-2620152757 */
          "prose",
          /** Deprecated since Tailwind CSS v4.0.0. @see https://github.com/tailwindlabs/tailwindcss.com/issues/2027#issuecomment-2620152757 */
          {
            screen: [i]
          },
          ...Y()
        ]
      }],
      /**
       * Height
       * @see https://tailwindcss.com/docs/height
       */
      h: [{
        h: ["screen", "lh", ...Y()]
      }],
      /**
       * Min-Height
       * @see https://tailwindcss.com/docs/min-height
       */
      "min-h": [{
        "min-h": ["screen", "lh", "none", ...Y()]
      }],
      /**
       * Max-Height
       * @see https://tailwindcss.com/docs/max-height
       */
      "max-h": [{
        "max-h": ["screen", "lh", ...Y()]
      }],
      // ------------------
      // --- Typography ---
      // ------------------
      /**
       * Font Size
       * @see https://tailwindcss.com/docs/font-size
       */
      "font-size": [{
        text: ["base", o, re, q]
      }],
      /**
       * Font Smoothing
       * @see https://tailwindcss.com/docs/font-smoothing
       */
      "font-smoothing": ["antialiased", "subpixel-antialiased"],
      /**
       * Font Style
       * @see https://tailwindcss.com/docs/font-style
       */
      "font-style": ["italic", "not-italic"],
      /**
       * Font Weight
       * @see https://tailwindcss.com/docs/font-weight
       */
      "font-weight": [{
        font: [n, $r, Er]
      }],
      /**
       * Font Stretch
       * @see https://tailwindcss.com/docs/font-stretch
       */
      "font-stretch": [{
        "font-stretch": ["ultra-condensed", "extra-condensed", "condensed", "semi-condensed", "normal", "semi-expanded", "expanded", "extra-expanded", "ultra-expanded", fe, m]
      }],
      /**
       * Font Family
       * @see https://tailwindcss.com/docs/font-family
       */
      "font-family": [{
        font: [Fr, Gr, r]
      }],
      /**
       * Font Feature Settings
       * @see https://tailwindcss.com/docs/font-feature-settings
       */
      "font-features": [{
        "font-features": [m]
      }],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-normal": ["normal-nums"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-ordinal": ["ordinal"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-slashed-zero": ["slashed-zero"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-figure": ["lining-nums", "oldstyle-nums"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-spacing": ["proportional-nums", "tabular-nums"],
      /**
       * Font Variant Numeric
       * @see https://tailwindcss.com/docs/font-variant-numeric
       */
      "fvn-fraction": ["diagonal-fractions", "stacked-fractions"],
      /**
       * Letter Spacing
       * @see https://tailwindcss.com/docs/letter-spacing
       */
      tracking: [{
        tracking: [s, f, m]
      }],
      /**
       * Line Clamp
       * @see https://tailwindcss.com/docs/line-clamp
       */
      "line-clamp": [{
        "line-clamp": [y, "none", f, Me]
      }],
      /**
       * Line Height
       * @see https://tailwindcss.com/docs/line-height
       */
      leading: [{
        leading: [
          /** Deprecated since Tailwind CSS v4.0.0. @see https://github.com/tailwindlabs/tailwindcss.com/issues/2027#issuecomment-2620152757 */
          a,
          ...x()
        ]
      }],
      /**
       * List Style Image
       * @see https://tailwindcss.com/docs/list-style-image
       */
      "list-image": [{
        "list-image": ["none", f, m]
      }],
      /**
       * List Style Position
       * @see https://tailwindcss.com/docs/list-style-position
       */
      "list-style-position": [{
        list: ["inside", "outside"]
      }],
      /**
       * List Style Type
       * @see https://tailwindcss.com/docs/list-style-type
       */
      "list-style-type": [{
        list: ["disc", "decimal", "none", f, m]
      }],
      /**
       * Text Alignment
       * @see https://tailwindcss.com/docs/text-align
       */
      "text-alignment": [{
        text: ["left", "center", "right", "justify", "start", "end"]
      }],
      /**
       * Placeholder Color
       * @deprecated since Tailwind CSS v3.0.0
       * @see https://v3.tailwindcss.com/docs/placeholder-color
       */
      "placeholder-color": [{
        placeholder: g()
      }],
      /**
       * Text Color
       * @see https://tailwindcss.com/docs/text-color
       */
      "text-color": [{
        text: g()
      }],
      /**
       * Text Decoration
       * @see https://tailwindcss.com/docs/text-decoration
       */
      "text-decoration": ["underline", "overline", "line-through", "no-underline"],
      /**
       * Text Decoration Style
       * @see https://tailwindcss.com/docs/text-decoration-style
       */
      "text-decoration-style": [{
        decoration: [...te(), "wavy"]
      }],
      /**
       * Text Decoration Thickness
       * @see https://tailwindcss.com/docs/text-decoration-thickness
       */
      "text-decoration-thickness": [{
        decoration: [y, "from-font", "auto", f, q]
      }],
      /**
       * Text Decoration Color
       * @see https://tailwindcss.com/docs/text-decoration-color
       */
      "text-decoration-color": [{
        decoration: g()
      }],
      /**
       * Text Underline Offset
       * @see https://tailwindcss.com/docs/text-underline-offset
       */
      "underline-offset": [{
        "underline-offset": [y, "auto", f, m]
      }],
      /**
       * Text Transform
       * @see https://tailwindcss.com/docs/text-transform
       */
      "text-transform": ["uppercase", "lowercase", "capitalize", "normal-case"],
      /**
       * Text Overflow
       * @see https://tailwindcss.com/docs/text-overflow
       */
      "text-overflow": ["truncate", "text-ellipsis", "text-clip"],
      /**
       * Text Wrap
       * @see https://tailwindcss.com/docs/text-wrap
       */
      "text-wrap": [{
        text: ["wrap", "nowrap", "balance", "pretty"]
      }],
      /**
       * Text Indent
       * @see https://tailwindcss.com/docs/text-indent
       */
      indent: [{
        indent: x()
      }],
      /**
       * Vertical Alignment
       * @see https://tailwindcss.com/docs/vertical-align
       */
      "vertical-align": [{
        align: ["baseline", "top", "middle", "bottom", "text-top", "text-bottom", "sub", "super", f, m]
      }],
      /**
       * Whitespace
       * @see https://tailwindcss.com/docs/whitespace
       */
      whitespace: [{
        whitespace: ["normal", "nowrap", "pre", "pre-line", "pre-wrap", "break-spaces"]
      }],
      /**
       * Word Break
       * @see https://tailwindcss.com/docs/word-break
       */
      break: [{
        break: ["normal", "words", "all", "keep"]
      }],
      /**
       * Overflow Wrap
       * @see https://tailwindcss.com/docs/overflow-wrap
       */
      wrap: [{
        wrap: ["break-word", "anywhere", "normal"]
      }],
      /**
       * Hyphens
       * @see https://tailwindcss.com/docs/hyphens
       */
      hyphens: [{
        hyphens: ["none", "manual", "auto"]
      }],
      /**
       * Content
       * @see https://tailwindcss.com/docs/content
       */
      content: [{
        content: ["none", f, m]
      }],
      // -------------------
      // --- Backgrounds ---
      // -------------------
      /**
       * Background Attachment
       * @see https://tailwindcss.com/docs/background-attachment
       */
      "bg-attachment": [{
        bg: ["fixed", "local", "scroll"]
      }],
      /**
       * Background Clip
       * @see https://tailwindcss.com/docs/background-clip
       */
      "bg-clip": [{
        "bg-clip": ["border", "padding", "content", "text"]
      }],
      /**
       * Background Origin
       * @see https://tailwindcss.com/docs/background-origin
       */
      "bg-origin": [{
        "bg-origin": ["border", "padding", "content"]
      }],
      /**
       * Background Position
       * @see https://tailwindcss.com/docs/background-position
       */
      "bg-position": [{
        bg: ve()
      }],
      /**
       * Background Repeat
       * @see https://tailwindcss.com/docs/background-repeat
       */
      "bg-repeat": [{
        bg: ye()
      }],
      /**
       * Background Size
       * @see https://tailwindcss.com/docs/background-size
       */
      "bg-size": [{
        bg: we()
      }],
      /**
       * Background Image
       * @see https://tailwindcss.com/docs/background-image
       */
      "bg-image": [{
        bg: ["none", {
          linear: [{
            to: ["t", "tr", "r", "br", "b", "bl", "l", "tl"]
          }, U, f, m],
          radial: ["", f, m],
          conic: [U, f, m]
        }, Dr, Or]
      }],
      /**
       * Background Color
       * @see https://tailwindcss.com/docs/background-color
       */
      "bg-color": [{
        bg: g()
      }],
      /**
       * Gradient Color Stops From Position
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-from-pos": [{
        from: pe()
      }],
      /**
       * Gradient Color Stops Via Position
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-via-pos": [{
        via: pe()
      }],
      /**
       * Gradient Color Stops To Position
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-to-pos": [{
        to: pe()
      }],
      /**
       * Gradient Color Stops From
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-from": [{
        from: g()
      }],
      /**
       * Gradient Color Stops Via
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-via": [{
        via: g()
      }],
      /**
       * Gradient Color Stops To
       * @see https://tailwindcss.com/docs/gradient-color-stops
       */
      "gradient-to": [{
        to: g()
      }],
      // ---------------
      // --- Borders ---
      // ---------------
      /**
       * Border Radius
       * @see https://tailwindcss.com/docs/border-radius
       */
      rounded: [{
        rounded: P()
      }],
      /**
       * Border Radius Start
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-s": [{
        "rounded-s": P()
      }],
      /**
       * Border Radius End
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-e": [{
        "rounded-e": P()
      }],
      /**
       * Border Radius Top
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-t": [{
        "rounded-t": P()
      }],
      /**
       * Border Radius Right
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-r": [{
        "rounded-r": P()
      }],
      /**
       * Border Radius Bottom
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-b": [{
        "rounded-b": P()
      }],
      /**
       * Border Radius Left
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-l": [{
        "rounded-l": P()
      }],
      /**
       * Border Radius Start Start
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-ss": [{
        "rounded-ss": P()
      }],
      /**
       * Border Radius Start End
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-se": [{
        "rounded-se": P()
      }],
      /**
       * Border Radius End End
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-ee": [{
        "rounded-ee": P()
      }],
      /**
       * Border Radius End Start
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-es": [{
        "rounded-es": P()
      }],
      /**
       * Border Radius Top Left
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-tl": [{
        "rounded-tl": P()
      }],
      /**
       * Border Radius Top Right
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-tr": [{
        "rounded-tr": P()
      }],
      /**
       * Border Radius Bottom Right
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-br": [{
        "rounded-br": P()
      }],
      /**
       * Border Radius Bottom Left
       * @see https://tailwindcss.com/docs/border-radius
       */
      "rounded-bl": [{
        "rounded-bl": P()
      }],
      /**
       * Border Width
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w": [{
        border: E()
      }],
      /**
       * Border Width Inline
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-x": [{
        "border-x": E()
      }],
      /**
       * Border Width Block
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-y": [{
        "border-y": E()
      }],
      /**
       * Border Width Inline Start
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-s": [{
        "border-s": E()
      }],
      /**
       * Border Width Inline End
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-e": [{
        "border-e": E()
      }],
      /**
       * Border Width Block Start
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-bs": [{
        "border-bs": E()
      }],
      /**
       * Border Width Block End
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-be": [{
        "border-be": E()
      }],
      /**
       * Border Width Top
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-t": [{
        "border-t": E()
      }],
      /**
       * Border Width Right
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-r": [{
        "border-r": E()
      }],
      /**
       * Border Width Bottom
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-b": [{
        "border-b": E()
      }],
      /**
       * Border Width Left
       * @see https://tailwindcss.com/docs/border-width
       */
      "border-w-l": [{
        "border-l": E()
      }],
      /**
       * Divide Width X
       * @see https://tailwindcss.com/docs/border-width#between-children
       */
      "divide-x": [{
        "divide-x": E()
      }],
      /**
       * Divide Width X Reverse
       * @see https://tailwindcss.com/docs/border-width#between-children
       */
      "divide-x-reverse": ["divide-x-reverse"],
      /**
       * Divide Width Y
       * @see https://tailwindcss.com/docs/border-width#between-children
       */
      "divide-y": [{
        "divide-y": E()
      }],
      /**
       * Divide Width Y Reverse
       * @see https://tailwindcss.com/docs/border-width#between-children
       */
      "divide-y-reverse": ["divide-y-reverse"],
      /**
       * Border Style
       * @see https://tailwindcss.com/docs/border-style
       */
      "border-style": [{
        border: [...te(), "hidden", "none"]
      }],
      /**
       * Divide Style
       * @see https://tailwindcss.com/docs/border-style#setting-the-divider-style
       */
      "divide-style": [{
        divide: [...te(), "hidden", "none"]
      }],
      /**
       * Border Color
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color": [{
        border: g()
      }],
      /**
       * Border Color Inline
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-x": [{
        "border-x": g()
      }],
      /**
       * Border Color Block
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-y": [{
        "border-y": g()
      }],
      /**
       * Border Color Inline Start
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-s": [{
        "border-s": g()
      }],
      /**
       * Border Color Inline End
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-e": [{
        "border-e": g()
      }],
      /**
       * Border Color Block Start
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-bs": [{
        "border-bs": g()
      }],
      /**
       * Border Color Block End
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-be": [{
        "border-be": g()
      }],
      /**
       * Border Color Top
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-t": [{
        "border-t": g()
      }],
      /**
       * Border Color Right
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-r": [{
        "border-r": g()
      }],
      /**
       * Border Color Bottom
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-b": [{
        "border-b": g()
      }],
      /**
       * Border Color Left
       * @see https://tailwindcss.com/docs/border-color
       */
      "border-color-l": [{
        "border-l": g()
      }],
      /**
       * Divide Color
       * @see https://tailwindcss.com/docs/divide-color
       */
      "divide-color": [{
        divide: g()
      }],
      /**
       * Outline Style
       * @see https://tailwindcss.com/docs/outline-style
       */
      "outline-style": [{
        outline: [...te(), "none", "hidden"]
      }],
      /**
       * Outline Offset
       * @see https://tailwindcss.com/docs/outline-offset
       */
      "outline-offset": [{
        "outline-offset": [y, f, m]
      }],
      /**
       * Outline Width
       * @see https://tailwindcss.com/docs/outline-width
       */
      "outline-w": [{
        outline: ["", y, re, q]
      }],
      /**
       * Outline Color
       * @see https://tailwindcss.com/docs/outline-color
       */
      "outline-color": [{
        outline: g()
      }],
      // ---------------
      // --- Effects ---
      // ---------------
      /**
       * Box Shadow
       * @see https://tailwindcss.com/docs/box-shadow
       */
      shadow: [{
        shadow: [
          // Deprecated since Tailwind CSS v4.0.0
          "",
          "none",
          d,
          ie,
          ae
        ]
      }],
      /**
       * Box Shadow Color
       * @see https://tailwindcss.com/docs/box-shadow#setting-the-shadow-color
       */
      "shadow-color": [{
        shadow: g()
      }],
      /**
       * Inset Box Shadow
       * @see https://tailwindcss.com/docs/box-shadow#adding-an-inset-shadow
       */
      "inset-shadow": [{
        "inset-shadow": ["none", b, ie, ae]
      }],
      /**
       * Inset Box Shadow Color
       * @see https://tailwindcss.com/docs/box-shadow#setting-the-inset-shadow-color
       */
      "inset-shadow-color": [{
        "inset-shadow": g()
      }],
      /**
       * Ring Width
       * @see https://tailwindcss.com/docs/box-shadow#adding-a-ring
       */
      "ring-w": [{
        ring: E()
      }],
      /**
       * Ring Width Inset
       * @see https://v3.tailwindcss.com/docs/ring-width#inset-rings
       * @deprecated since Tailwind CSS v4.0.0
       * @see https://github.com/tailwindlabs/tailwindcss/blob/v4.0.0/packages/tailwindcss/src/utilities.ts#L4158
       */
      "ring-w-inset": ["ring-inset"],
      /**
       * Ring Color
       * @see https://tailwindcss.com/docs/box-shadow#setting-the-ring-color
       */
      "ring-color": [{
        ring: g()
      }],
      /**
       * Ring Offset Width
       * @see https://v3.tailwindcss.com/docs/ring-offset-width
       * @deprecated since Tailwind CSS v4.0.0
       * @see https://github.com/tailwindlabs/tailwindcss/blob/v4.0.0/packages/tailwindcss/src/utilities.ts#L4158
       */
      "ring-offset-w": [{
        "ring-offset": [y, q]
      }],
      /**
       * Ring Offset Color
       * @see https://v3.tailwindcss.com/docs/ring-offset-color
       * @deprecated since Tailwind CSS v4.0.0
       * @see https://github.com/tailwindlabs/tailwindcss/blob/v4.0.0/packages/tailwindcss/src/utilities.ts#L4158
       */
      "ring-offset-color": [{
        "ring-offset": g()
      }],
      /**
       * Inset Ring Width
       * @see https://tailwindcss.com/docs/box-shadow#adding-an-inset-ring
       */
      "inset-ring-w": [{
        "inset-ring": E()
      }],
      /**
       * Inset Ring Color
       * @see https://tailwindcss.com/docs/box-shadow#setting-the-inset-ring-color
       */
      "inset-ring-color": [{
        "inset-ring": g()
      }],
      /**
       * Text Shadow
       * @see https://tailwindcss.com/docs/text-shadow
       */
      "text-shadow": [{
        "text-shadow": ["none", w, ie, ae]
      }],
      /**
       * Text Shadow Color
       * @see https://tailwindcss.com/docs/text-shadow#setting-the-shadow-color
       */
      "text-shadow-color": [{
        "text-shadow": g()
      }],
      /**
       * Opacity
       * @see https://tailwindcss.com/docs/opacity
       */
      opacity: [{
        opacity: [y, f, m]
      }],
      /**
       * Mix Blend Mode
       * @see https://tailwindcss.com/docs/mix-blend-mode
       */
      "mix-blend": [{
        "mix-blend": [...ke(), "plus-darker", "plus-lighter"]
      }],
      /**
       * Background Blend Mode
       * @see https://tailwindcss.com/docs/background-blend-mode
       */
      "bg-blend": [{
        "bg-blend": ke()
      }],
      /**
       * Mask Clip
       * @see https://tailwindcss.com/docs/mask-clip
       */
      "mask-clip": [{
        "mask-clip": ["border", "padding", "content", "fill", "stroke", "view"]
      }, "mask-no-clip"],
      /**
       * Mask Composite
       * @see https://tailwindcss.com/docs/mask-composite
       */
      "mask-composite": [{
        mask: ["add", "subtract", "intersect", "exclude"]
      }],
      /**
       * Mask Image
       * @see https://tailwindcss.com/docs/mask-image
       */
      "mask-image-linear-pos": [{
        "mask-linear": [y]
      }],
      "mask-image-linear-from-pos": [{
        "mask-linear-from": B()
      }],
      "mask-image-linear-to-pos": [{
        "mask-linear-to": B()
      }],
      "mask-image-linear-from-color": [{
        "mask-linear-from": g()
      }],
      "mask-image-linear-to-color": [{
        "mask-linear-to": g()
      }],
      "mask-image-t-from-pos": [{
        "mask-t-from": B()
      }],
      "mask-image-t-to-pos": [{
        "mask-t-to": B()
      }],
      "mask-image-t-from-color": [{
        "mask-t-from": g()
      }],
      "mask-image-t-to-color": [{
        "mask-t-to": g()
      }],
      "mask-image-r-from-pos": [{
        "mask-r-from": B()
      }],
      "mask-image-r-to-pos": [{
        "mask-r-to": B()
      }],
      "mask-image-r-from-color": [{
        "mask-r-from": g()
      }],
      "mask-image-r-to-color": [{
        "mask-r-to": g()
      }],
      "mask-image-b-from-pos": [{
        "mask-b-from": B()
      }],
      "mask-image-b-to-pos": [{
        "mask-b-to": B()
      }],
      "mask-image-b-from-color": [{
        "mask-b-from": g()
      }],
      "mask-image-b-to-color": [{
        "mask-b-to": g()
      }],
      "mask-image-l-from-pos": [{
        "mask-l-from": B()
      }],
      "mask-image-l-to-pos": [{
        "mask-l-to": B()
      }],
      "mask-image-l-from-color": [{
        "mask-l-from": g()
      }],
      "mask-image-l-to-color": [{
        "mask-l-to": g()
      }],
      "mask-image-x-from-pos": [{
        "mask-x-from": B()
      }],
      "mask-image-x-to-pos": [{
        "mask-x-to": B()
      }],
      "mask-image-x-from-color": [{
        "mask-x-from": g()
      }],
      "mask-image-x-to-color": [{
        "mask-x-to": g()
      }],
      "mask-image-y-from-pos": [{
        "mask-y-from": B()
      }],
      "mask-image-y-to-pos": [{
        "mask-y-to": B()
      }],
      "mask-image-y-from-color": [{
        "mask-y-from": g()
      }],
      "mask-image-y-to-color": [{
        "mask-y-to": g()
      }],
      "mask-image-radial": [{
        "mask-radial": [f, m]
      }],
      "mask-image-radial-from-pos": [{
        "mask-radial-from": B()
      }],
      "mask-image-radial-to-pos": [{
        "mask-radial-to": B()
      }],
      "mask-image-radial-from-color": [{
        "mask-radial-from": g()
      }],
      "mask-image-radial-to-color": [{
        "mask-radial-to": g()
      }],
      "mask-image-radial-shape": [{
        "mask-radial": ["circle", "ellipse"]
      }],
      "mask-image-radial-size": [{
        "mask-radial": [{
          closest: ["side", "corner"],
          farthest: ["side", "corner"]
        }]
      }],
      "mask-image-radial-pos": [{
        "mask-radial-at": N()
      }],
      "mask-image-conic-pos": [{
        "mask-conic": [y]
      }],
      "mask-image-conic-from-pos": [{
        "mask-conic-from": B()
      }],
      "mask-image-conic-to-pos": [{
        "mask-conic-to": B()
      }],
      "mask-image-conic-from-color": [{
        "mask-conic-from": g()
      }],
      "mask-image-conic-to-color": [{
        "mask-conic-to": g()
      }],
      /**
       * Mask Mode
       * @see https://tailwindcss.com/docs/mask-mode
       */
      "mask-mode": [{
        mask: ["alpha", "luminance", "match"]
      }],
      /**
       * Mask Origin
       * @see https://tailwindcss.com/docs/mask-origin
       */
      "mask-origin": [{
        "mask-origin": ["border", "padding", "content", "fill", "stroke", "view"]
      }],
      /**
       * Mask Position
       * @see https://tailwindcss.com/docs/mask-position
       */
      "mask-position": [{
        mask: ve()
      }],
      /**
       * Mask Repeat
       * @see https://tailwindcss.com/docs/mask-repeat
       */
      "mask-repeat": [{
        mask: ye()
      }],
      /**
       * Mask Size
       * @see https://tailwindcss.com/docs/mask-size
       */
      "mask-size": [{
        mask: we()
      }],
      /**
       * Mask Type
       * @see https://tailwindcss.com/docs/mask-type
       */
      "mask-type": [{
        "mask-type": ["alpha", "luminance"]
      }],
      /**
       * Mask Image
       * @see https://tailwindcss.com/docs/mask-image
       */
      "mask-image": [{
        mask: ["none", f, m]
      }],
      // ---------------
      // --- Filters ---
      // ---------------
      /**
       * Filter
       * @see https://tailwindcss.com/docs/filter
       */
      filter: [{
        filter: [
          // Deprecated since Tailwind CSS v3.0.0
          "",
          "none",
          f,
          m
        ]
      }],
      /**
       * Blur
       * @see https://tailwindcss.com/docs/blur
       */
      blur: [{
        blur: Ce()
      }],
      /**
       * Brightness
       * @see https://tailwindcss.com/docs/brightness
       */
      brightness: [{
        brightness: [y, f, m]
      }],
      /**
       * Contrast
       * @see https://tailwindcss.com/docs/contrast
       */
      contrast: [{
        contrast: [y, f, m]
      }],
      /**
       * Drop Shadow
       * @see https://tailwindcss.com/docs/drop-shadow
       */
      "drop-shadow": [{
        "drop-shadow": [
          // Deprecated since Tailwind CSS v4.0.0
          "",
          "none",
          v,
          ie,
          ae
        ]
      }],
      /**
       * Drop Shadow Color
       * @see https://tailwindcss.com/docs/filter-drop-shadow#setting-the-shadow-color
       */
      "drop-shadow-color": [{
        "drop-shadow": g()
      }],
      /**
       * Grayscale
       * @see https://tailwindcss.com/docs/grayscale
       */
      grayscale: [{
        grayscale: ["", y, f, m]
      }],
      /**
       * Hue Rotate
       * @see https://tailwindcss.com/docs/hue-rotate
       */
      "hue-rotate": [{
        "hue-rotate": [y, f, m]
      }],
      /**
       * Invert
       * @see https://tailwindcss.com/docs/invert
       */
      invert: [{
        invert: ["", y, f, m]
      }],
      /**
       * Saturate
       * @see https://tailwindcss.com/docs/saturate
       */
      saturate: [{
        saturate: [y, f, m]
      }],
      /**
       * Sepia
       * @see https://tailwindcss.com/docs/sepia
       */
      sepia: [{
        sepia: ["", y, f, m]
      }],
      /**
       * Backdrop Filter
       * @see https://tailwindcss.com/docs/backdrop-filter
       */
      "backdrop-filter": [{
        "backdrop-filter": [
          // Deprecated since Tailwind CSS v3.0.0
          "",
          "none",
          f,
          m
        ]
      }],
      /**
       * Backdrop Blur
       * @see https://tailwindcss.com/docs/backdrop-blur
       */
      "backdrop-blur": [{
        "backdrop-blur": Ce()
      }],
      /**
       * Backdrop Brightness
       * @see https://tailwindcss.com/docs/backdrop-brightness
       */
      "backdrop-brightness": [{
        "backdrop-brightness": [y, f, m]
      }],
      /**
       * Backdrop Contrast
       * @see https://tailwindcss.com/docs/backdrop-contrast
       */
      "backdrop-contrast": [{
        "backdrop-contrast": [y, f, m]
      }],
      /**
       * Backdrop Grayscale
       * @see https://tailwindcss.com/docs/backdrop-grayscale
       */
      "backdrop-grayscale": [{
        "backdrop-grayscale": ["", y, f, m]
      }],
      /**
       * Backdrop Hue Rotate
       * @see https://tailwindcss.com/docs/backdrop-hue-rotate
       */
      "backdrop-hue-rotate": [{
        "backdrop-hue-rotate": [y, f, m]
      }],
      /**
       * Backdrop Invert
       * @see https://tailwindcss.com/docs/backdrop-invert
       */
      "backdrop-invert": [{
        "backdrop-invert": ["", y, f, m]
      }],
      /**
       * Backdrop Opacity
       * @see https://tailwindcss.com/docs/backdrop-opacity
       */
      "backdrop-opacity": [{
        "backdrop-opacity": [y, f, m]
      }],
      /**
       * Backdrop Saturate
       * @see https://tailwindcss.com/docs/backdrop-saturate
       */
      "backdrop-saturate": [{
        "backdrop-saturate": [y, f, m]
      }],
      /**
       * Backdrop Sepia
       * @see https://tailwindcss.com/docs/backdrop-sepia
       */
      "backdrop-sepia": [{
        "backdrop-sepia": ["", y, f, m]
      }],
      // --------------
      // --- Tables ---
      // --------------
      /**
       * Border Collapse
       * @see https://tailwindcss.com/docs/border-collapse
       */
      "border-collapse": [{
        border: ["collapse", "separate"]
      }],
      /**
       * Border Spacing
       * @see https://tailwindcss.com/docs/border-spacing
       */
      "border-spacing": [{
        "border-spacing": x()
      }],
      /**
       * Border Spacing X
       * @see https://tailwindcss.com/docs/border-spacing
       */
      "border-spacing-x": [{
        "border-spacing-x": x()
      }],
      /**
       * Border Spacing Y
       * @see https://tailwindcss.com/docs/border-spacing
       */
      "border-spacing-y": [{
        "border-spacing-y": x()
      }],
      /**
       * Table Layout
       * @see https://tailwindcss.com/docs/table-layout
       */
      "table-layout": [{
        table: ["auto", "fixed"]
      }],
      /**
       * Caption Side
       * @see https://tailwindcss.com/docs/caption-side
       */
      caption: [{
        caption: ["top", "bottom"]
      }],
      // ---------------------------------
      // --- Transitions and Animation ---
      // ---------------------------------
      /**
       * Transition Property
       * @see https://tailwindcss.com/docs/transition-property
       */
      transition: [{
        transition: ["", "all", "colors", "opacity", "shadow", "transform", "none", f, m]
      }],
      /**
       * Transition Behavior
       * @see https://tailwindcss.com/docs/transition-behavior
       */
      "transition-behavior": [{
        transition: ["normal", "discrete"]
      }],
      /**
       * Transition Duration
       * @see https://tailwindcss.com/docs/transition-duration
       */
      duration: [{
        duration: [y, "initial", f, m]
      }],
      /**
       * Transition Timing Function
       * @see https://tailwindcss.com/docs/transition-timing-function
       */
      ease: [{
        ease: ["linear", "initial", A, f, m]
      }],
      /**
       * Transition Delay
       * @see https://tailwindcss.com/docs/transition-delay
       */
      delay: [{
        delay: [y, f, m]
      }],
      /**
       * Animation
       * @see https://tailwindcss.com/docs/animation
       */
      animate: [{
        animate: ["none", j, f, m]
      }],
      // ------------------
      // --- Transforms ---
      // ------------------
      /**
       * Backface Visibility
       * @see https://tailwindcss.com/docs/backface-visibility
       */
      backface: [{
        backface: ["hidden", "visible"]
      }],
      /**
       * Perspective
       * @see https://tailwindcss.com/docs/perspective
       */
      perspective: [{
        perspective: [C, f, m]
      }],
      /**
       * Perspective Origin
       * @see https://tailwindcss.com/docs/perspective-origin
       */
      "perspective-origin": [{
        "perspective-origin": I()
      }],
      /**
       * Rotate
       * @see https://tailwindcss.com/docs/rotate
       */
      rotate: [{
        rotate: oe()
      }],
      /**
       * Rotate X
       * @see https://tailwindcss.com/docs/rotate
       */
      "rotate-x": [{
        "rotate-x": oe()
      }],
      /**
       * Rotate Y
       * @see https://tailwindcss.com/docs/rotate
       */
      "rotate-y": [{
        "rotate-y": oe()
      }],
      /**
       * Rotate Z
       * @see https://tailwindcss.com/docs/rotate
       */
      "rotate-z": [{
        "rotate-z": oe()
      }],
      /**
       * Scale
       * @see https://tailwindcss.com/docs/scale
       */
      scale: [{
        scale: ne()
      }],
      /**
       * Scale X
       * @see https://tailwindcss.com/docs/scale
       */
      "scale-x": [{
        "scale-x": ne()
      }],
      /**
       * Scale Y
       * @see https://tailwindcss.com/docs/scale
       */
      "scale-y": [{
        "scale-y": ne()
      }],
      /**
       * Scale Z
       * @see https://tailwindcss.com/docs/scale
       */
      "scale-z": [{
        "scale-z": ne()
      }],
      /**
       * Scale 3D
       * @see https://tailwindcss.com/docs/scale
       */
      "scale-3d": ["scale-3d"],
      /**
       * Skew
       * @see https://tailwindcss.com/docs/skew
       */
      skew: [{
        skew: me()
      }],
      /**
       * Skew X
       * @see https://tailwindcss.com/docs/skew
       */
      "skew-x": [{
        "skew-x": me()
      }],
      /**
       * Skew Y
       * @see https://tailwindcss.com/docs/skew
       */
      "skew-y": [{
        "skew-y": me()
      }],
      /**
       * Transform
       * @see https://tailwindcss.com/docs/transform
       */
      transform: [{
        transform: [f, m, "", "none", "gpu", "cpu"]
      }],
      /**
       * Transform Origin
       * @see https://tailwindcss.com/docs/transform-origin
       */
      "transform-origin": [{
        origin: I()
      }],
      /**
       * Transform Style
       * @see https://tailwindcss.com/docs/transform-style
       */
      "transform-style": [{
        transform: ["3d", "flat"]
      }],
      /**
       * Translate
       * @see https://tailwindcss.com/docs/translate
       */
      translate: [{
        translate: se()
      }],
      /**
       * Translate X
       * @see https://tailwindcss.com/docs/translate
       */
      "translate-x": [{
        "translate-x": se()
      }],
      /**
       * Translate Y
       * @see https://tailwindcss.com/docs/translate
       */
      "translate-y": [{
        "translate-y": se()
      }],
      /**
       * Translate Z
       * @see https://tailwindcss.com/docs/translate
       */
      "translate-z": [{
        "translate-z": se()
      }],
      /**
       * Translate None
       * @see https://tailwindcss.com/docs/translate
       */
      "translate-none": ["translate-none"],
      // ---------------------
      // --- Interactivity ---
      // ---------------------
      /**
       * Accent Color
       * @see https://tailwindcss.com/docs/accent-color
       */
      accent: [{
        accent: g()
      }],
      /**
       * Appearance
       * @see https://tailwindcss.com/docs/appearance
       */
      appearance: [{
        appearance: ["none", "auto"]
      }],
      /**
       * Caret Color
       * @see https://tailwindcss.com/docs/just-in-time-mode#caret-color-utilities
       */
      "caret-color": [{
        caret: g()
      }],
      /**
       * Color Scheme
       * @see https://tailwindcss.com/docs/color-scheme
       */
      "color-scheme": [{
        scheme: ["normal", "dark", "light", "light-dark", "only-dark", "only-light"]
      }],
      /**
       * Cursor
       * @see https://tailwindcss.com/docs/cursor
       */
      cursor: [{
        cursor: ["auto", "default", "pointer", "wait", "text", "move", "help", "not-allowed", "none", "context-menu", "progress", "cell", "crosshair", "vertical-text", "alias", "copy", "no-drop", "grab", "grabbing", "all-scroll", "col-resize", "row-resize", "n-resize", "e-resize", "s-resize", "w-resize", "ne-resize", "nw-resize", "se-resize", "sw-resize", "ew-resize", "ns-resize", "nesw-resize", "nwse-resize", "zoom-in", "zoom-out", f, m]
      }],
      /**
       * Field Sizing
       * @see https://tailwindcss.com/docs/field-sizing
       */
      "field-sizing": [{
        "field-sizing": ["fixed", "content"]
      }],
      /**
       * Pointer Events
       * @see https://tailwindcss.com/docs/pointer-events
       */
      "pointer-events": [{
        "pointer-events": ["auto", "none"]
      }],
      /**
       * Resize
       * @see https://tailwindcss.com/docs/resize
       */
      resize: [{
        resize: ["none", "", "y", "x"]
      }],
      /**
       * Scroll Behavior
       * @see https://tailwindcss.com/docs/scroll-behavior
       */
      "scroll-behavior": [{
        scroll: ["auto", "smooth"]
      }],
      /**
       * Scroll Margin
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-m": [{
        "scroll-m": x()
      }],
      /**
       * Scroll Margin Inline
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mx": [{
        "scroll-mx": x()
      }],
      /**
       * Scroll Margin Block
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-my": [{
        "scroll-my": x()
      }],
      /**
       * Scroll Margin Inline Start
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-ms": [{
        "scroll-ms": x()
      }],
      /**
       * Scroll Margin Inline End
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-me": [{
        "scroll-me": x()
      }],
      /**
       * Scroll Margin Block Start
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mbs": [{
        "scroll-mbs": x()
      }],
      /**
       * Scroll Margin Block End
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mbe": [{
        "scroll-mbe": x()
      }],
      /**
       * Scroll Margin Top
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mt": [{
        "scroll-mt": x()
      }],
      /**
       * Scroll Margin Right
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mr": [{
        "scroll-mr": x()
      }],
      /**
       * Scroll Margin Bottom
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-mb": [{
        "scroll-mb": x()
      }],
      /**
       * Scroll Margin Left
       * @see https://tailwindcss.com/docs/scroll-margin
       */
      "scroll-ml": [{
        "scroll-ml": x()
      }],
      /**
       * Scroll Padding
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-p": [{
        "scroll-p": x()
      }],
      /**
       * Scroll Padding Inline
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-px": [{
        "scroll-px": x()
      }],
      /**
       * Scroll Padding Block
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-py": [{
        "scroll-py": x()
      }],
      /**
       * Scroll Padding Inline Start
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-ps": [{
        "scroll-ps": x()
      }],
      /**
       * Scroll Padding Inline End
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pe": [{
        "scroll-pe": x()
      }],
      /**
       * Scroll Padding Block Start
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pbs": [{
        "scroll-pbs": x()
      }],
      /**
       * Scroll Padding Block End
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pbe": [{
        "scroll-pbe": x()
      }],
      /**
       * Scroll Padding Top
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pt": [{
        "scroll-pt": x()
      }],
      /**
       * Scroll Padding Right
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pr": [{
        "scroll-pr": x()
      }],
      /**
       * Scroll Padding Bottom
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pb": [{
        "scroll-pb": x()
      }],
      /**
       * Scroll Padding Left
       * @see https://tailwindcss.com/docs/scroll-padding
       */
      "scroll-pl": [{
        "scroll-pl": x()
      }],
      /**
       * Scroll Snap Align
       * @see https://tailwindcss.com/docs/scroll-snap-align
       */
      "snap-align": [{
        snap: ["start", "end", "center", "align-none"]
      }],
      /**
       * Scroll Snap Stop
       * @see https://tailwindcss.com/docs/scroll-snap-stop
       */
      "snap-stop": [{
        snap: ["normal", "always"]
      }],
      /**
       * Scroll Snap Type
       * @see https://tailwindcss.com/docs/scroll-snap-type
       */
      "snap-type": [{
        snap: ["none", "x", "y", "both"]
      }],
      /**
       * Scroll Snap Type Strictness
       * @see https://tailwindcss.com/docs/scroll-snap-type
       */
      "snap-strictness": [{
        snap: ["mandatory", "proximity"]
      }],
      /**
       * Touch Action
       * @see https://tailwindcss.com/docs/touch-action
       */
      touch: [{
        touch: ["auto", "none", "manipulation"]
      }],
      /**
       * Touch Action X
       * @see https://tailwindcss.com/docs/touch-action
       */
      "touch-x": [{
        "touch-pan": ["x", "left", "right"]
      }],
      /**
       * Touch Action Y
       * @see https://tailwindcss.com/docs/touch-action
       */
      "touch-y": [{
        "touch-pan": ["y", "up", "down"]
      }],
      /**
       * Touch Action Pinch Zoom
       * @see https://tailwindcss.com/docs/touch-action
       */
      "touch-pz": ["touch-pinch-zoom"],
      /**
       * User Select
       * @see https://tailwindcss.com/docs/user-select
       */
      select: [{
        select: ["none", "text", "all", "auto"]
      }],
      /**
       * Will Change
       * @see https://tailwindcss.com/docs/will-change
       */
      "will-change": [{
        "will-change": ["auto", "scroll", "contents", "transform", f, m]
      }],
      // -----------
      // --- SVG ---
      // -----------
      /**
       * Fill
       * @see https://tailwindcss.com/docs/fill
       */
      fill: [{
        fill: ["none", ...g()]
      }],
      /**
       * Stroke Width
       * @see https://tailwindcss.com/docs/stroke-width
       */
      "stroke-w": [{
        stroke: [y, re, q, Me]
      }],
      /**
       * Stroke
       * @see https://tailwindcss.com/docs/stroke
       */
      stroke: [{
        stroke: ["none", ...g()]
      }],
      // ---------------------
      // --- Accessibility ---
      // ---------------------
      /**
       * Forced Color Adjust
       * @see https://tailwindcss.com/docs/forced-color-adjust
       */
      "forced-color-adjust": [{
        "forced-color-adjust": ["auto", "none"]
      }]
    },
    conflictingClassGroups: {
      overflow: ["overflow-x", "overflow-y"],
      overscroll: ["overscroll-x", "overscroll-y"],
      inset: ["inset-x", "inset-y", "inset-bs", "inset-be", "start", "end", "top", "right", "bottom", "left"],
      "inset-x": ["right", "left"],
      "inset-y": ["top", "bottom"],
      flex: ["basis", "grow", "shrink"],
      gap: ["gap-x", "gap-y"],
      p: ["px", "py", "ps", "pe", "pbs", "pbe", "pt", "pr", "pb", "pl"],
      px: ["pr", "pl"],
      py: ["pt", "pb"],
      m: ["mx", "my", "ms", "me", "mbs", "mbe", "mt", "mr", "mb", "ml"],
      mx: ["mr", "ml"],
      my: ["mt", "mb"],
      size: ["w", "h"],
      "font-size": ["leading"],
      "fvn-normal": ["fvn-ordinal", "fvn-slashed-zero", "fvn-figure", "fvn-spacing", "fvn-fraction"],
      "fvn-ordinal": ["fvn-normal"],
      "fvn-slashed-zero": ["fvn-normal"],
      "fvn-figure": ["fvn-normal"],
      "fvn-spacing": ["fvn-normal"],
      "fvn-fraction": ["fvn-normal"],
      "line-clamp": ["display", "overflow"],
      rounded: ["rounded-s", "rounded-e", "rounded-t", "rounded-r", "rounded-b", "rounded-l", "rounded-ss", "rounded-se", "rounded-ee", "rounded-es", "rounded-tl", "rounded-tr", "rounded-br", "rounded-bl"],
      "rounded-s": ["rounded-ss", "rounded-es"],
      "rounded-e": ["rounded-se", "rounded-ee"],
      "rounded-t": ["rounded-tl", "rounded-tr"],
      "rounded-r": ["rounded-tr", "rounded-br"],
      "rounded-b": ["rounded-br", "rounded-bl"],
      "rounded-l": ["rounded-tl", "rounded-bl"],
      "border-spacing": ["border-spacing-x", "border-spacing-y"],
      "border-w": ["border-w-x", "border-w-y", "border-w-s", "border-w-e", "border-w-bs", "border-w-be", "border-w-t", "border-w-r", "border-w-b", "border-w-l"],
      "border-w-x": ["border-w-r", "border-w-l"],
      "border-w-y": ["border-w-t", "border-w-b"],
      "border-color": ["border-color-x", "border-color-y", "border-color-s", "border-color-e", "border-color-bs", "border-color-be", "border-color-t", "border-color-r", "border-color-b", "border-color-l"],
      "border-color-x": ["border-color-r", "border-color-l"],
      "border-color-y": ["border-color-t", "border-color-b"],
      translate: ["translate-x", "translate-y", "translate-none"],
      "translate-none": ["translate", "translate-x", "translate-y", "translate-z"],
      "scroll-m": ["scroll-mx", "scroll-my", "scroll-ms", "scroll-me", "scroll-mbs", "scroll-mbe", "scroll-mt", "scroll-mr", "scroll-mb", "scroll-ml"],
      "scroll-mx": ["scroll-mr", "scroll-ml"],
      "scroll-my": ["scroll-mt", "scroll-mb"],
      "scroll-p": ["scroll-px", "scroll-py", "scroll-ps", "scroll-pe", "scroll-pbs", "scroll-pbe", "scroll-pt", "scroll-pr", "scroll-pb", "scroll-pl"],
      "scroll-px": ["scroll-pr", "scroll-pl"],
      "scroll-py": ["scroll-pt", "scroll-pb"],
      touch: ["touch-x", "touch-y", "touch-pz"],
      "touch-x": ["touch"],
      "touch-y": ["touch"],
      "touch-pz": ["touch"]
    },
    conflictingClassGroupModifiers: {
      "font-size": ["leading"]
    },
    orderSensitiveModifiers: ["*", "**", "after", "backdrop", "before", "details-content", "file", "first-letter", "first-line", "marker", "placeholder", "selection"]
  };
}, Zr = /* @__PURE__ */ Lr(Ur);
function h(...e) {
  return Zr(nr(e));
}
function Yr({ size: e = 32, className: r }) {
  return /* @__PURE__ */ t(
    "div",
    {
      className: h("relative shrink-0", r),
      style: { width: e, height: e },
      children: /* @__PURE__ */ p(
        "svg",
        {
          className: "block size-full",
          viewBox: "0 0 32 32",
          fill: "none",
          role: "img",
          "aria-label": "Astra logo",
          children: [
            /* @__PURE__ */ t(
              "path",
              {
                d: "M0 12.8C0 8.31958 0 6.07937 0.871948 4.36808C1.63893 2.86278 2.86278 1.63893 4.36808 0.871948C6.07937 0 8.31958 0 12.8 0H19.2C23.6804 0 25.9206 0 27.6319 0.871948C29.1372 1.63893 30.3611 2.86278 31.1281 4.36808C32 6.07937 32 8.31958 32 12.8V19.2C32 23.6804 32 25.9206 31.1281 27.6319C30.3611 29.1372 29.1372 30.3611 27.6319 31.1281C25.9206 32 23.6804 32 19.2 32H12.8C8.31958 32 6.07937 32 4.36808 31.1281C2.86278 30.3611 1.63893 29.1372 0.871948 27.6319C0 25.9206 0 23.6804 0 19.2V12.8Z",
                fill: "var(--brand-primary)"
              }
            ),
            /* @__PURE__ */ t(
              "path",
              {
                d: "M16.3472 3.99756C17.3609 11.2801 20.7163 14.635 27.9985 15.6489V16.3452C20.7158 17.359 17.3608 20.7154 16.3472 27.9985H15.6509C14.6371 20.715 11.2813 17.3588 3.99756 16.3452V15.6489C11.2808 14.6352 14.6369 11.2805 15.6509 3.99756H16.3472Z",
                fill: "white"
              }
            )
          ]
        }
      )
    }
  );
}
function qr({
  type: e = "image",
  size: r = "large",
  shape: o = "circle",
  initials: n = "F",
  src: s,
  alt: a = "Avatar",
  onClick: i,
  className: l
}) {
  const c = {
    small: "size-[24px]",
    medium: "size-[32px]",
    large: "size-[40px]"
  }, u = {
    circle: "rounded-[999px]",
    square: "rounded-[8px]"
  }, d = h(
    c[r],
    u[o],
    "relative overflow-hidden",
    i && "cursor-pointer hover:opacity-90 transition-opacity",
    l
  );
  return e === "initial" ? /* @__PURE__ */ t(
    "div",
    {
      className: h(d, "bg-brand-primary flex items-center justify-center"),
      onClick: i,
      children: /* @__PURE__ */ t("div", { className: "text-white font-medium text-center leading-none font-sans", children: /* @__PURE__ */ t("p", { className: "text-label-sm", children: n }) })
    }
  ) : /* @__PURE__ */ p("div", { className: d, onClick: i, children: [
    s && /* @__PURE__ */ t(
      "img",
      {
        alt: a,
        className: "absolute inset-0 w-full h-full object-cover pointer-events-none",
        src: s
      }
    ),
    /* @__PURE__ */ t(
      "div",
      {
        "aria-hidden": "true",
        className: h(
          "absolute inset-0 border border-border-primary pointer-events-none",
          u[o]
        )
      }
    )
  ] });
}
function Ie({ children: e, size: r = "md", onClick: o }) {
  const n = {
    sm: "size-[32px]",
    md: "size-[40px]",
    lg: "size-[48px]"
  };
  return /* @__PURE__ */ p(
    "div",
    {
      className: h(
        "relative rounded-full shrink-0 cursor-pointer transition-transform hover:scale-105",
        n[r],
        o && "cursor-pointer"
      ),
      onClick: o,
      children: [
        /* @__PURE__ */ t("div", { className: h("overflow-hidden relative rounded-full", n[r]), children: /* @__PURE__ */ t("div", { className: h("absolute left-1/2 top-1/2 translate-x-[-50%] translate-y-[-50%]", n[r]), children: e }) }),
        /* @__PURE__ */ t(
          "div",
          {
            "aria-hidden": "true",
            className: "absolute border border-border-primary inset-0 pointer-events-none rounded-full"
          }
        )
      ]
    }
  );
}
function Se({
  count: e,
  size: r = "md",
  onClick: o
}) {
  return /* @__PURE__ */ t(
    "div",
    {
      className: h(
        "bg-brand-primary flex items-center justify-center rounded-full shrink-0 font-medium transition-colors hover:bg-brand-hover cursor-pointer",
        {
          sm: "size-[20px] text-[10px]",
          md: "size-[24px] text-label-sm",
          lg: "size-[28px] text-label"
        }[r],
        o && "cursor-pointer"
      ),
      onClick: o,
      children: /* @__PURE__ */ p("span", { className: "leading-none text-white", children: [
        "+",
        e
      ] })
    }
  );
}
function ft({
  avatars: e,
  maxVisible: r = 3,
  spacing: o = "spaced",
  showOverflow: n = !0,
  overflowCount: s,
  size: a = "md",
  onAvatarClick: i,
  onOverflowClick: l,
  className: c
}) {
  const u = e.slice(0, r), d = s ?? Math.max(0, e.length - r), b = n && d > 0;
  return o === "overlap" ? /* @__PURE__ */ p("div", { className: h("flex items-center isolate", c), children: [
    /* @__PURE__ */ t("div", { className: "flex items-center relative", children: u.map((w, v) => /* @__PURE__ */ t("div", { className: h("relative", v > 0 && "-ml-2"), style: { zIndex: u.length - v }, children: /* @__PURE__ */ t(
      Ie,
      {
        size: a,
        onClick: i ? () => i(v) : void 0,
        children: /* @__PURE__ */ t(
          "img",
          {
            alt: w.alt || `Avatar ${v + 1}`,
            className: "absolute inset-0 w-full h-full object-cover pointer-events-none",
            src: w.src
          }
        )
      }
    ) }, v)) }),
    b && /* @__PURE__ */ t("div", { className: "ml-2", children: /* @__PURE__ */ t(
      Se,
      {
        count: d,
        size: a,
        onClick: l
      }
    ) })
  ] }) : /* @__PURE__ */ p("div", { className: h("flex items-center gap-2", c), children: [
    u.map((w, v) => /* @__PURE__ */ t(
      Ie,
      {
        size: a,
        onClick: i ? () => i(v) : void 0,
        children: /* @__PURE__ */ t(
          "img",
          {
            alt: w.alt || `Avatar ${v + 1}`,
            className: "absolute inset-0 w-full h-full object-cover pointer-events-none",
            src: w.src
          }
        )
      },
      v
    )),
    b && /* @__PURE__ */ t(
      Se,
      {
        count: d,
        size: a,
        onClick: l
      }
    )
  ] });
}
function qe({
  label: e = "Label",
  variant: r = "default",
  removable: o = !1,
  onRemove: n,
  className: s
}) {
  const a = {
    default: "bg-brand-tertiary text-text-primary",
    success: "bg-success text-surface-darkest",
    warning: "bg-warning text-surface-darkest",
    danger: "bg-danger text-surface-darkest",
    brand: "bg-brand-primary text-white",
    secondary: "bg-brand-secondary text-text-primary"
  }, i = {
    default: "var(--text-primary)",
    success: "var(--surface-darkest)",
    warning: "var(--surface-darkest)",
    danger: "var(--surface-darkest)",
    brand: "white",
    secondary: "var(--text-primary)"
  }, l = () => {
    n && n();
  };
  return /* @__PURE__ */ p("div", { className: h(
    "box-border flex items-center gap-[8px] overflow-hidden px-[8px] py-[6px] rounded-[8px] w-fit",
    a[r],
    s
  ), children: [
    /* @__PURE__ */ t("span", { className: "font-medium text-label-sm leading-none shrink-0", children: e }),
    o && /* @__PURE__ */ t(
      "button",
      {
        onClick: l,
        className: "relative shrink-0 size-[6px] cursor-pointer hover:opacity-70 transition-opacity",
        "aria-label": "Remove badge",
        children: /* @__PURE__ */ t("div", { className: "absolute inset-[-12.5%]", children: /* @__PURE__ */ t(
          "svg",
          {
            className: "block size-full",
            fill: "none",
            preserveAspectRatio: "none",
            viewBox: "0 0 8 8",
            children: /* @__PURE__ */ t(
              "path",
              {
                d: "M7 1L1 7M1 1L7 7",
                stroke: i[r],
                strokeLinecap: "round",
                strokeLinejoin: "round",
                strokeWidth: "1.5"
              }
            )
          }
        ) })
      }
    )
  ] });
}
function Kr({
  icon: e,
  variant: r = "primary",
  size: o = "medium",
  disabled: n = !1,
  className: s,
  ...a
}) {
  const i = "inline-flex items-center justify-center rounded-full transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-brand-primary cursor-pointer disabled:opacity-20 disabled:pointer-events-none", l = {
    primary: "bg-brand-primary text-white hover:bg-brand-secondary",
    neutral: "bg-transparent text-brand-primary border border-brand-primary hover:bg-brand-secondary hover:border-brand-secondary",
    subtle: "bg-transparent text-text-secondary hover:bg-bg-hover"
  }, c = {
    medium: "size-[40px] p-3",
    small: "size-[32px] p-2"
  };
  return /* @__PURE__ */ t(
    "button",
    {
      className: h(i, l[r], c[o], s),
      disabled: n,
      ...a,
      children: /* @__PURE__ */ t("span", { className: "shrink-0", children: e })
    }
  );
}
/**
 * @license lucide-react v0.577.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const Ke = (...e) => e.filter((r, o, n) => !!r && r.trim() !== "" && n.indexOf(r) === o).join(" ").trim();
/**
 * @license lucide-react v0.577.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const Xr = (e) => e.replace(/([a-z0-9])([A-Z])/g, "$1-$2").toLowerCase();
/**
 * @license lucide-react v0.577.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const Jr = (e) => e.replace(
  /^([A-Z])|[\s-_]+(\w)/g,
  (r, o, n) => n ? n.toUpperCase() : o.toLowerCase()
);
/**
 * @license lucide-react v0.577.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const Re = (e) => {
  const r = Jr(e);
  return r.charAt(0).toUpperCase() + r.slice(1);
};
/**
 * @license lucide-react v0.577.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
var Qr = {
  xmlns: "http://www.w3.org/2000/svg",
  width: 24,
  height: 24,
  viewBox: "0 0 24 24",
  fill: "none",
  stroke: "currentColor",
  strokeWidth: 2,
  strokeLinecap: "round",
  strokeLinejoin: "round"
};
/**
 * @license lucide-react v0.577.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const et = (e) => {
  for (const r in e)
    if (r.startsWith("aria-") || r === "role" || r === "title")
      return !0;
  return !1;
};
/**
 * @license lucide-react v0.577.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const rt = _e(
  ({
    color: e = "currentColor",
    size: r = 24,
    strokeWidth: o = 2,
    absoluteStrokeWidth: n,
    className: s = "",
    children: a,
    iconNode: i,
    ...l
  }, c) => he(
    "svg",
    {
      ref: c,
      ...Qr,
      width: r,
      height: r,
      stroke: e,
      strokeWidth: n ? Number(o) * 24 / Number(r) : o,
      className: Ke("lucide", s),
      ...!a && !et(l) && { "aria-hidden": "true" },
      ...l
    },
    [
      ...i.map(([u, d]) => he(u, d)),
      ...Array.isArray(a) ? a : [a]
    ]
  )
);
/**
 * @license lucide-react v0.577.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const Xe = (e, r) => {
  const o = _e(
    ({ className: n, ...s }, a) => he(rt, {
      ref: a,
      iconNode: r,
      className: Ke(
        `lucide-${Xr(Re(e))}`,
        `lucide-${e}`,
        n
      ),
      ...s
    })
  );
  return o.displayName = Re(e), o;
};
/**
 * @license lucide-react v0.577.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const tt = [
  ["path", { d: "m21 21-4.34-4.34", key: "14j7rj" }],
  ["circle", { cx: "11", cy: "11", r: "8", key: "4ej97u" }]
], ot = Xe("search", tt);
/**
 * @license lucide-react v0.577.0 - ISC
 *
 * This source code is licensed under the ISC license.
 * See the LICENSE file in the root directory of this source tree.
 */
const nt = [
  [
    "path",
    {
      d: "M14.536 21.686a.5.5 0 0 0 .937-.024l6.5-19a.496.496 0 0 0-.635-.635l-19 6.5a.5.5 0 0 0-.024.937l7.93 3.18a2 2 0 0 1 1.112 1.11z",
      key: "1ffxy3"
    }
  ],
  ["path", { d: "m21.854 2.147-10.94 10.939", key: "12cjpa" }]
], st = Xe("send", nt);
function Je({
  value: e = "",
  placeholder: r = "Describe your video",
  onChange: o,
  onSend: n,
  onAttach: s,
  disabled: a = !1,
  className: i
}) {
  const [l, c] = R(e), [u, d] = R([]), b = (C) => {
    const k = C.target.value;
    c(k), o == null || o(k);
  }, w = () => {
    !a && l.trim() && (n == null || n());
  }, v = () => {
    if (!a) {
      const C = `Attachment ${u.length + 1}`;
      d((k) => [...k, C]), s == null || s();
    }
  }, M = (C) => {
    d((k) => k.filter((A, j) => j !== C));
  };
  return /* @__PURE__ */ p(
    "div",
    {
      className: h(
        "bg-surface-bg box-border content-stretch flex flex-col items-start justify-between p-[24px] rounded-[24px] w-full relative transition-all duration-300",
        u.length > 0 ? "h-[200px]" : "h-[135px]",
        a && "opacity-50 pointer-events-none",
        i
      ),
      children: [
        /* @__PURE__ */ t("div", { className: "absolute border border-border-primary border-solid inset-0 pointer-events-none rounded-[24px]" }),
        u.length > 0 && /* @__PURE__ */ t("div", { className: "flex flex-wrap gap-2 w-full mb-2", children: u.map((C, k) => /* @__PURE__ */ t(
          qe,
          {
            label: C,
            variant: "brand",
            removable: !0,
            onRemove: () => M(k)
          },
          k
        )) }),
        /* @__PURE__ */ t(
          Kr,
          {
            icon: /* @__PURE__ */ t(st, { size: 16 }),
            variant: "primary",
            size: "medium",
            onClick: w,
            disabled: a || !l.trim(),
            className: "absolute bottom-[10px] right-[10px] hover:scale-105 transition-transform"
          }
        ),
        /* @__PURE__ */ t(
          "button",
          {
            onClick: v,
            disabled: a,
            className: "absolute left-[24px] bottom-[12px] overflow-clip size-[20px] hover:scale-110 transition-transform disabled:opacity-50 disabled:cursor-not-allowed focus:outline-none focus:ring-2 focus:ring-brand-primary focus:ring-offset-2 rounded",
            children: /* @__PURE__ */ t("div", { className: "absolute inset-[5.78%_10.67%_8.34%_8.34%]", children: /* @__PURE__ */ t("div", { className: "absolute inset-[-4.37%_-4.63%]", children: /* @__PURE__ */ t("svg", { className: "block size-full", fill: "none", preserveAspectRatio: "none", viewBox: "0 0 18 19", children: /* @__PURE__ */ t(
              "path",
              {
                d: "M17.1986 9.05185L9.54028 16.7102C8.60208 17.6484 7.3296 18.1755 6.00278 18.1755C4.67596 18.1755 3.40348 17.6484 2.46528 16.7102C1.52708 15.772 1 14.4995 1 13.1727C1 11.8459 1.52708 10.5734 2.46528 9.63519L10.1236 1.97685C10.7491 1.35138 11.5974 1 12.4819 1C13.3665 1 14.2148 1.35138 14.8403 1.97685C15.4657 2.60232 15.8171 3.45064 15.8171 4.33519C15.8171 5.21973 15.4657 6.06805 14.8403 6.69352L7.17361 14.3519C6.86088 14.6646 6.43672 14.8403 5.99445 14.8403C5.55217 14.8403 5.12801 14.6646 4.81528 14.3519C4.50255 14.0391 4.32685 13.615 4.32685 13.1727C4.32685 12.7304 4.50255 12.3063 4.81528 11.9935L11.8903 4.92685",
                stroke: "var(--text-primary)",
                strokeLinecap: "round",
                strokeLinejoin: "round",
                strokeWidth: "1.5"
              }
            ) }) }) })
          }
        ),
        /* @__PURE__ */ t(
          "textarea",
          {
            value: l,
            onChange: b,
            placeholder: r,
            disabled: a,
            className: "font-sans font-normal min-w-full relative shrink-0 text-input-sm text-text-primary w-full h-full bg-transparent border-none outline-none resize-none placeholder:text-text-primary pr-[60px] pb-[40px]"
          }
        )
      ]
    }
  );
}
const at = [
  { label: "Sizzle reel", prompt: "A Sizzle Reel of a design conference" },
  { label: "Documentary", prompt: "A Documentary about a design conference" },
  { label: "Ad", prompt: "An Ad for a design conference" }
];
function ht({
  value: e,
  placeholder: r,
  onChange: o,
  onSend: n,
  onAttach: s,
  suggestions: a = at,
  onSuggestionClick: i,
  disabled: l,
  className: c
}) {
  const [u, d] = R(e ?? ""), b = (v) => {
    d(v), o == null || o(v);
  }, w = (v) => {
    d(v.prompt), o == null || o(v.prompt), i == null || i(v);
  };
  return /* @__PURE__ */ p(
    "div",
    {
      className: h(
        "bg-brand-secondary border border-border-primary flex flex-col gap-4 items-start p-6 rounded-2xl w-full",
        c
      ),
      children: [
        /* @__PURE__ */ t("p", { className: "font-sans font-medium text-heading text-text-primary", children: "✦ Create a new video" }),
        /* @__PURE__ */ t(
          Je,
          {
            value: u,
            placeholder: r,
            onChange: b,
            onSend: n,
            onAttach: s,
            disabled: l,
            className: "w-full"
          }
        ),
        a.length > 0 && /* @__PURE__ */ t("div", { className: "flex flex-wrap gap-3 items-center", children: a.map((v) => /* @__PURE__ */ t(
          "button",
          {
            onClick: () => w(v),
            disabled: l,
            className: "focus:outline-none disabled:pointer-events-none",
            children: /* @__PURE__ */ t(qe, { label: v.label, variant: "brand" })
          },
          v.label
        )) })
      ]
    }
  );
}
function bt({
  variant: e = "primary",
  size: r = "medium",
  iconStart: o,
  iconEnd: n,
  children: s,
  disabled: a = !1,
  className: i,
  onClick: l,
  ...c
}) {
  const u = "inline-flex items-center justify-center gap-2 rounded-full font-sans leading-none transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-brand-primary cursor-pointer hover:cursor-pointer", d = {
    primary: "bg-brand-primary text-white border border-brand-primary hover:bg-brand-secondary hover:border-brand-secondary disabled:opacity-40 disabled:pointer-events-none",
    neutral: "bg-transparent text-brand-primary border border-brand-primary hover:text-brand-secondary hover:border-brand-secondary disabled:opacity-40 disabled:pointer-events-none",
    subtle: "bg-transparent text-brand-primary border-none hover:text-brand-secondary disabled:opacity-40 disabled:pointer-events-none"
  }, b = {
    medium: r === "medium" && e === "subtle" ? "h-10 px-3 py-1.5 text-base" : "h-10 px-4 py-2.5 text-base",
    small: "h-8 p-2 text-xs"
  };
  return /* @__PURE__ */ p(
    "button",
    {
      className: h(
        u,
        d[e],
        b[r],
        a && "cursor-not-allowed",
        i
      ),
      disabled: a,
      onClick: l,
      ...c,
      children: [
        o && /* @__PURE__ */ t("span", { className: "shrink-0", children: o }),
        /* @__PURE__ */ t("span", { className: "shrink-0", children: s }),
        n && /* @__PURE__ */ t("span", { className: "shrink-0", children: n })
      ]
    }
  );
}
function xt({
  children: e,
  align: r = "justify",
  className: o
}) {
  return /* @__PURE__ */ t(
    "div",
    {
      className: h(
        "flex gap-3 w-full",
        {
          justify: "flex-row justify-between",
          start: "flex-row justify-start",
          end: "flex-row justify-end",
          center: "flex-row justify-center",
          stack: "flex-col"
        }[r],
        r === "stack" && "[&>*]:w-full",
        r === "justify" && "[&>*]:flex-1",
        o
      ),
      children: e
    }
  );
}
function it() {
  return /* @__PURE__ */ t("div", { className: "relative shrink-0 size-[24px]", children: /* @__PURE__ */ t("svg", { className: "block size-full", fill: "none", preserveAspectRatio: "none", viewBox: "0 0 24 24", children: /* @__PURE__ */ p("g", { children: [
    /* @__PURE__ */ t("circle", { cx: "12", cy: "12", fill: "var(--brand-primary)", r: "12" }),
    /* @__PURE__ */ t("path", { d: "M12 6.58301C12.371 8.04326 12.9716 9.21687 13.877 10.1221C14.7824 11.0273 15.9565 11.6282 17.417 11.999C15.9565 12.37 14.7824 12.9706 13.877 13.876C12.9715 14.7815 12.371 15.9563 12 17.417C11.629 15.9563 11.0285 14.7815 10.123 13.876C9.21741 12.9705 8.0429 12.37 6.58203 11.999C8.04288 11.6281 9.21738 11.0275 10.123 10.1221C11.0284 9.21688 11.629 8.04324 12 6.58301Z", stroke: "white" })
  ] }) }) });
}
function gt({ type: e = "ai", text: r, userAvatar: o, className: n }) {
  return e === "user" ? /* @__PURE__ */ t("div", { className: h("box-border content-stretch flex flex-col gap-[8px] items-start pl-0 pr-[32px] py-0 w-[254px]", n), children: /* @__PURE__ */ t("div", { className: "content-stretch flex gap-[8px] items-start relative shrink-0", children: /* @__PURE__ */ p("div", { className: "box-border content-stretch flex gap-[8px] items-start pl-[24px] pr-0 py-0 relative shrink-0 w-[254px]", children: [
    /* @__PURE__ */ p("div", { className: "bg-surface-bg flex-[1_0_0] min-h-px min-w-px relative rounded-[8px] shrink-0", children: [
      /* @__PURE__ */ t("div", { "aria-hidden": "true", className: "absolute border border-border-secondary border-solid inset-0 pointer-events-none rounded-[8px]" }),
      /* @__PURE__ */ t("div", { className: "size-full", children: /* @__PURE__ */ t("div", { className: "box-border content-stretch flex gap-[12px] items-start px-[16px] py-[12px] relative w-full", children: /* @__PURE__ */ t("div", { className: "content-stretch flex flex-[1_0_0] flex-col gap-[10px] items-center justify-center min-h-px min-w-px relative shrink-0", children: /* @__PURE__ */ t("p", { className: "font-sans font-normal relative shrink-0 text-video-title text-text-primary w-full whitespace-pre-wrap", children: r }) }) }) })
    ] }),
    /* @__PURE__ */ t("div", { className: "box-border content-stretch flex gap-[10px] items-center pb-0 pt-[6px] px-0 relative shrink-0", children: /* @__PURE__ */ t("div", { className: "relative rounded-full shrink-0 size-[24px]", children: o || /* @__PURE__ */ t(qr, { type: "initial", initials: "U", size: "small", shape: "circle" }) }) })
  ] }) }) }) : /* @__PURE__ */ t("div", { className: h("box-border content-stretch flex flex-col gap-[8px] items-start pl-0 pr-[32px] py-0 w-[254px]", n), children: /* @__PURE__ */ p("div", { className: "content-stretch flex gap-[8px] items-start relative shrink-0", children: [
    /* @__PURE__ */ t("div", { className: "box-border content-stretch flex gap-[10px] items-center pb-0 pt-[6px] px-0 relative shrink-0", children: /* @__PURE__ */ t("div", { className: "content-stretch flex items-start relative shrink-0", children: /* @__PURE__ */ t(it, {}) }) }),
    /* @__PURE__ */ p("div", { className: "bg-brand-secondary box-border content-stretch flex gap-[12px] items-start px-[16px] py-[12px] relative rounded-[8px] shrink-0 w-[212px]", children: [
      /* @__PURE__ */ t("div", { "aria-hidden": "true", className: "absolute border border-border-secondary border-solid inset-0 pointer-events-none rounded-[8px]" }),
      /* @__PURE__ */ t("div", { className: "content-stretch flex flex-[1_0_0] gap-[10px] items-center justify-center min-h-px min-w-px relative shrink-0", children: /* @__PURE__ */ t("p", { className: "flex-[1_0_0] font-sans font-normal min-h-px min-w-px relative shrink-0 text-video-title text-text-primary whitespace-pre-wrap", children: r }) })
    ] })
  ] }) });
}
function vt({
  children: e,
  value: r,
  placeholder: o,
  onChange: n,
  onSend: s,
  onAttach: a,
  disabled: i,
  className: l
}) {
  return /* @__PURE__ */ p(
    "div",
    {
      className: h(
        "bg-[rgba(255,255,255,0.5)] dark:bg-[rgba(255,255,255,0.05)] border-r border-border-primary flex flex-col gap-4 px-4 py-6 h-full w-[400px]",
        l
      ),
      children: [
        /* @__PURE__ */ t("div", { className: "flex-1 flex flex-col gap-3 min-h-0 overflow-y-auto p-2.5 [&>*]:w-full", children: e }),
        /* @__PURE__ */ t("div", { className: "shrink-0 w-full", children: /* @__PURE__ */ t(
          Je,
          {
            value: r,
            placeholder: o,
            onChange: n,
            onSend: s,
            onAttach: a,
            disabled: i
          }
        ) })
      ]
    }
  );
}
function yt({
  value: e = "",
  onChange: r,
  onSearch: o,
  placeholder: n = "Search",
  disabled: s = !1,
  className: a,
  ...i
}) {
  const l = ["anything", "clips", "audio"], [c, u] = R(0), [d, b] = R(0), w = (n == null ? void 0 : n.trim()) || "Search", v = l[c], M = `${w} ${v.slice(0, d)}`.trim();
  F(() => {
    if (e && e.length > 0) return;
    const A = 100, j = 2e3;
    let T;
    return d < v.length ? T = setTimeout(() => b((N) => N + 1), A) : T = setTimeout(() => {
      b(0), u((N) => (N + 1) % l.length);
    }, j), () => clearTimeout(T);
  }, [d, v, e]);
  const C = (A) => {
    A.key === "Enter" && o && o(e);
  }, k = (A) => {
    const j = A.target.value;
    r == null || r(j);
  };
  return /* @__PURE__ */ t("div", { className: h("content-stretch flex gap-[8px] items-center w-full", a), children: /* @__PURE__ */ p("div", { className: "bg-surface-bg box-border content-stretch flex gap-[8px] h-[40px] items-center px-[20px] py-[12px] relative rounded-[64px] w-full focus-within:ring-2 focus-within:ring-border-primary transition-all", children: [
    /* @__PURE__ */ t(ot, { size: 16, className: "shrink-0 text-text-secondary" }),
    /* @__PURE__ */ t(
      "input",
      {
        type: "text",
        value: e,
        onChange: k,
        onKeyDown: C,
        placeholder: e ? n : M,
        disabled: s,
        className: "bg-transparent font-sans font-normal text-input-sm text-text-primary placeholder:text-text-secondary border-none outline-none w-full",
        ...i
      }
    )
  ] }) });
}
function wt({
  segments: e,
  selectedSegment: r,
  onChange: o,
  className: n
}) {
  return /* @__PURE__ */ t(
    "div",
    {
      className: h(
        "bg-bg-faint flex gap-1 items-center px-2 py-1 rounded-[14px] h-14 w-fit",
        n
      ),
      children: e.map((s) => {
        const a = s.id === r;
        return /* @__PURE__ */ t(
          "button",
          {
            onClick: () => o(s.id),
            className: h(
              "flex items-center justify-center p-3 rounded-lg size-12 transition-colors",
              a ? "bg-brand-muted" : "hover:bg-surface-dark-hover"
            ),
            children: /* @__PURE__ */ t(
              "div",
              {
                className: h(
                  "size-6 flex items-center justify-center",
                  a ? "text-white" : "text-text-secondary"
                ),
                children: s.icon
              }
            )
          },
          s.id
        );
      })
    }
  );
}
function kt({
  value: e,
  placeholder: r = "I am a placeholder...",
  label: o,
  description: n,
  prefix: s,
  suffix: a,
  disabled: i = !1,
  className: l,
  onChange: c,
  ...u
}) {
  const d = X(null);
  return /* @__PURE__ */ t("div", { className: h("relative", l), children: /* @__PURE__ */ p("div", { className: "flex flex-col gap-1.5", children: [
    o && /* @__PURE__ */ t(
      "label",
      {
        className: "text-label-sm font-medium text-text-primary cursor-pointer",
        onClick: () => {
          var b;
          return (b = d.current) == null ? void 0 : b.focus();
        },
        children: o
      }
    ),
    /* @__PURE__ */ p(
      "div",
      {
        className: h(
          "flex items-center gap-2 bg-bg-subtle w-full rounded-lg px-3 py-2.5 transition-colors",
          "has-[:focus]:bg-bg-hover",
          !i && "hover:bg-bg-hover",
          i && "opacity-50 cursor-not-allowed"
        ),
        onClick: () => {
          var b;
          return (b = d.current) == null ? void 0 : b.focus();
        },
        children: [
          s && /* @__PURE__ */ t("span", { className: "shrink-0 text-text-secondary", children: s }),
          /* @__PURE__ */ t(
            "input",
            {
              ref: d,
              type: "text",
              value: e,
              placeholder: r,
              disabled: i,
              onChange: (b) => c == null ? void 0 : c(b.target.value),
              className: h(
                "bg-transparent w-full font-sans text-input-sm leading-none font-normal text-text-primary",
                "placeholder:text-text-secondary",
                "outline-none border-none"
              ),
              ...u
            }
          ),
          a && /* @__PURE__ */ t("span", { className: "shrink-0 text-text-secondary", children: a })
        ]
      }
    ),
    n && /* @__PURE__ */ t("span", { className: "text-video-title text-text-secondary", children: n })
  ] }) });
}
function lt({
  duration: e,
  className: r
}) {
  return /* @__PURE__ */ t(
    "div",
    {
      className: h(
        "inline-flex items-center px-2 py-1.5 rounded-lg bg-input-bg overflow-clip",
        r
      ),
      children: /* @__PURE__ */ t("span", { className: "text-input-sm whitespace-nowrap", style: { color: "var(--on-brand)" }, children: e })
    }
  );
}
function Ct({
  title: e = "Item Title",
  updated: r = "Edited 2m ago",
  spec: o = "4K",
  duration: n = "0:01:30",
  thumbnail: s,
  className: a
}) {
  return /* @__PURE__ */ p(
    "div",
    {
      className: h(
        "flex flex-col items-start overflow-clip rounded-lg border border-border-primary bg-surface-bg w-[305px]",
        a
      ),
      children: [
        /* @__PURE__ */ p(
          "div",
          {
            className: "relative w-full aspect-[316/177] overflow-clip",
            style: s ? void 0 : {
              backgroundImage: "linear-gradient(168deg, rgb(26, 58, 42) 8%, rgb(45, 90, 58) 42%, rgb(74, 122, 74) 92%)"
            },
            children: [
              s && /* @__PURE__ */ t("div", { className: "absolute inset-0", children: s }),
              n && /* @__PURE__ */ t(lt, { duration: n, className: "absolute bottom-3 right-3 z-10" })
            ]
          }
        ),
        /* @__PURE__ */ p("div", { className: "flex flex-col gap-1 items-start w-full px-3 py-3", children: [
          /* @__PURE__ */ t("p", { className: "text-label text-text-primary w-full", children: e }),
          /* @__PURE__ */ p("div", { className: "flex items-center gap-1.5 h-[13px]", children: [
            /* @__PURE__ */ t("span", { className: "text-video-title text-text-primary whitespace-nowrap", children: r }),
            /* @__PURE__ */ t("div", { className: "size-[3px] rounded-sm bg-on-reverse shrink-0" }),
            /* @__PURE__ */ t("span", { className: "text-video-title text-text-primary whitespace-nowrap", children: o })
          ] })
        ] })
      ]
    }
  );
}
function Nt({
  options: e,
  value: r = "",
  onChange: o,
  placeholder: n = "Select an option",
  label: s,
  description: a,
  state: i = "default",
  disabled: l = !1,
  className: c
}) {
  const [u, d] = R(!1), [b, w] = R(!1), [v, M] = R(-1), C = X(null), k = X(null), A = X([]), j = e.find((L) => L.value === r), T = j ? j.label : n, N = !j, I = 5, V = 36, G = (L) => {
    L < 0 || requestAnimationFrame(() => {
      const z = A.current[L];
      z == null || z.scrollIntoView({ block: "nearest" });
    });
  }, x = () => {
    if (l) return;
    w(!1), d(!0);
    const L = Math.max(
      e.findIndex((z) => z.value === r),
      0
    );
    M(L), G(L);
  }, _ = () => {
    if (!u) return;
    w(!0);
    const L = setTimeout(() => {
      d(!1), w(!1), M(-1);
    }, 180);
    return () => clearTimeout(L);
  };
  F(() => {
    const L = (z) => {
      C.current && !C.current.contains(z.target) && _();
    };
    return document.addEventListener("mousedown", L), () => document.removeEventListener("mousedown", L);
  }, [u]), F(() => {
    const L = (z) => {
      if (!l) {
        if (!u) {
          (z.key === "Enter" || z.key === " ") && (z.preventDefault(), x());
          return;
        }
        switch (z.key) {
          case "Escape":
            z.preventDefault(), _();
            break;
          case "ArrowDown":
            z.preventDefault(), M(($) => {
              const W = Math.min($ + 1, e.length - 1);
              return G(W), W;
            });
            break;
          case "ArrowUp":
            z.preventDefault(), M(($) => {
              const W = Math.max($ - 1, 0);
              return G(W), W;
            });
            break;
          case "Enter":
          case " ":
            z.preventDefault(), v >= 0 && v < e.length && ee(e[v].value);
            break;
        }
      }
    };
    return document.addEventListener("keydown", L), () => document.removeEventListener("keydown", L);
  }, [u, v, e, l]);
  const ee = (L) => {
    o(L), _();
  };
  return /* @__PURE__ */ p("div", { className: h("relative", c), ref: C, children: [
    /* @__PURE__ */ p("div", { className: "flex flex-col gap-1.5", children: [
      s && /* @__PURE__ */ t("span", { className: "text-label-sm font-medium text-text-primary", children: s }),
      /* @__PURE__ */ p(
        "div",
        {
          className: h(
            "bg-bg-subtle flex items-center gap-2 rounded-lg px-3 py-2.5 cursor-pointer transition-colors",
            !l && "hover:bg-bg-hover",
            l && "opacity-50 cursor-not-allowed"
          ),
          onClick: () => u ? _() : x(),
          tabIndex: l ? -1 : 0,
          role: "combobox",
          "aria-expanded": u,
          "aria-haspopup": "listbox",
          "aria-controls": u ? "select-list" : void 0,
          "aria-activedescendant": v >= 0 ? `select-opt-${v}` : void 0,
          children: [
            /* @__PURE__ */ t(
              "span",
              {
                className: h(
                  "flex-1 font-sans text-input-sm leading-none",
                  N ? "text-text-secondary font-normal" : "text-text-primary font-normal"
                ),
                children: T
              }
            ),
            /* @__PURE__ */ t(
              "svg",
              {
                className: h(
                  "shrink-0 text-text-secondary transition-transform duration-200",
                  u && "rotate-180"
                ),
                width: "16",
                height: "16",
                viewBox: "0 0 16 16",
                fill: "none",
                children: /* @__PURE__ */ t(
                  "path",
                  {
                    d: "M4 6L8 10L12 6",
                    stroke: "currentColor",
                    strokeWidth: "1.5",
                    strokeLinecap: "round",
                    strokeLinejoin: "round"
                  }
                )
              }
            )
          ]
        }
      ),
      a && /* @__PURE__ */ t("span", { className: "text-video-title text-text-secondary", children: a })
    ] }),
    (u || b) && !l && /* @__PURE__ */ t(
      "div",
      {
        ref: k,
        id: "select-list",
        className: h(
          "absolute top-full left-0 right-0 mt-1 bg-surface-bg border border-border-primary rounded-lg shadow-lg z-50 overflow-y-auto",
          "transition-[opacity,transform] duration-200",
          b ? "opacity-0 -translate-y-1" : "opacity-100 translate-y-0"
        ),
        role: "listbox",
        style: {
          maxHeight: `${V * Math.min(I, e.length)}px`
        },
        "data-state": b ? "closing" : "open",
        children: e.map((L, z) => /* @__PURE__ */ t(
          "div",
          {
            id: `select-opt-${z}`,
            className: h(
              "px-3 py-2 cursor-pointer text-input-sm font-sans transition-colors",
              "hover:bg-bg-hover",
              v === z && "bg-bg-hover",
              r === L.value && "bg-brand-tertiary text-brand-primary font-medium"
            ),
            onClick: () => ee(L.value),
            onMouseEnter: () => M(z),
            role: "option",
            "aria-selected": r === L.value,
            ref: ($) => {
              A.current[z] = $;
            },
            children: L.label
          },
          L.value
        ))
      }
    )
  ] });
}
function Lt({
  icon: e,
  active: r = !1,
  className: o,
  ...n
}) {
  return /* @__PURE__ */ t(
    "button",
    {
      className: h(
        "flex items-center justify-center p-3 rounded-lg transition-colors cursor-pointer",
        r ? "bg-brand-tertiary text-text-primary" : "text-text-secondary hover:bg-bg-hover",
        o
      ),
      "aria-pressed": r,
      ...n,
      children: /* @__PURE__ */ t("span", { className: h("size-6", r ? "opacity-85" : "opacity-50"), children: e })
    }
  );
}
function zt({
  children: e,
  footer: r,
  className: o
}) {
  return /* @__PURE__ */ p(
    "nav",
    {
      className: h(
        "bg-surface-bg h-full w-[60px] border-r border-border-primary flex flex-col items-center py-3",
        o
      ),
      children: [
        /* @__PURE__ */ t("div", { className: "shrink-0", children: /* @__PURE__ */ t(Yr, { size: 32 }) }),
        /* @__PURE__ */ t("div", { className: "flex flex-col items-center gap-2 mt-3 w-full", children: e }),
        r && /* @__PURE__ */ t("div", { className: "mt-auto flex flex-col items-center gap-2 w-full", children: r })
      ]
    }
  );
}
function Mt({
  label: e = "Label",
  description: r = "Description",
  hasDescription: o = !0,
  showLabel: n = !0,
  defaultSelected: s = !0,
  onChange: a,
  disabled: i = !1
}) {
  const [l, c] = R(s);
  return /* @__PURE__ */ p("div", { className: "flex gap-4 items-start w-[360px]", children: [
    /* @__PURE__ */ t("div", { className: "flex flex-1 flex-col gap-2 min-h-0 min-w-0", children: /* @__PURE__ */ p("div", { className: "flex flex-col font-medium gap-4 leading-[1.4] text-base w-full", children: [
      n && /* @__PURE__ */ t("p", { className: "text-text-primary", children: e }),
      o && /* @__PURE__ */ t("p", { className: "text-text-secondary min-w-full whitespace-pre-wrap", children: r })
    ] }) }),
    /* @__PURE__ */ t(
      "button",
      {
        onClick: () => {
          if (i) return;
          const d = !l;
          c(d), a == null || a(d);
        },
        disabled: i,
        className: h(
          "flex items-center h-[18.389px] w-[31.995px] rounded-[25811100px] transition-all duration-200",
          "focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-brand-primary",
          "disabled:opacity-50 disabled:cursor-not-allowed",
          l ? "bg-brand-primary pl-[14.767px] pr-[0.769px] py-[0.769px]" : "bg-bg-subtle px-[1.769px] py-[0.769px]"
        ),
        children: /* @__PURE__ */ t("div", { className: "bg-white rounded-[25811100px] size-[15.998px] transition-transform duration-200" })
      }
    )
  ] });
}
function At({
  value: e,
  placeholder: r = "Your text goes here...",
  label: o,
  description: n,
  disabled: s = !1,
  className: a,
  onChange: i,
  rows: l = 3,
  ...c
}) {
  const u = X(null);
  return /* @__PURE__ */ t("div", { className: h("relative", a), children: /* @__PURE__ */ p("div", { className: "flex flex-col gap-1.5", children: [
    o && /* @__PURE__ */ t(
      "label",
      {
        className: "text-label-sm font-medium text-text-primary cursor-pointer",
        onClick: () => {
          var d;
          return (d = u.current) == null ? void 0 : d.focus();
        },
        children: o
      }
    ),
    /* @__PURE__ */ t(
      "textarea",
      {
        ref: u,
        value: e,
        placeholder: r,
        disabled: s,
        rows: l,
        onChange: (d) => i == null ? void 0 : i(d.target.value),
        className: h(
          "bg-bg-subtle w-full rounded-lg px-3 py-2.5 font-sans text-input-sm font-normal text-text-primary transition-colors resize-vertical",
          "placeholder:text-text-secondary",
          "outline-none border-none",
          "focus:bg-bg-hover",
          !s && "hover:bg-bg-hover",
          s && "opacity-50 cursor-not-allowed resize-none"
        ),
        ...c
      }
    ),
    n && /* @__PURE__ */ t("span", { className: "text-video-title text-text-secondary", children: n })
  ] }) });
}
function jt({
  message: e,
  progress: r = 0,
  showCancel: o = !0,
  onCancel: n,
  variant: s = "default",
  onDismiss: a
}) {
  const [i, l] = R(0), [c, u] = R(!1), d = r <= 0;
  F(() => {
    if (!d) return;
    const C = Date.now(), k = 7500, A = () => {
      const I = (Date.now() - C) % k / k * 100;
      l((V) => (I < V && (u(!0), setTimeout(() => u(!1), 0)), I)), j = requestAnimationFrame(A);
    };
    let j = requestAnimationFrame(A);
    return () => cancelAnimationFrame(j);
  }, [d]);
  const b = {
    default: "bg-brand-primary",
    success: "bg-success",
    error: "bg-danger",
    warning: "bg-warning"
  }, w = {
    default: "bg-brand-dark",
    success: "bg-success brightness-[0.85]",
    error: "bg-danger brightness-[0.85]",
    warning: "bg-warning brightness-[0.85]"
  }, v = Math.min(Math.max(r, 0), 100), M = d ? i : v;
  return /* @__PURE__ */ p(
    "div",
    {
      className: h(
        "box-border content-stretch flex gap-[6.772px] h-[34px] items-center min-h-[33.862px] overflow-clip pl-[10.159px] pr-[6.772px] py-[6.772px] rounded-[10.737px] shadow-[0px_0px_0.423px_0px_rgba(0,0,0,0.3),0px_0.847px_2.54px_0px_rgba(0,0,0,0.15)] relative",
        b[s]
      ),
      children: [
        /* @__PURE__ */ t(
          "div",
          {
            className: h(
              "absolute h-[33.862px] left-0 top-0 rounded-l-[10.737px]",
              // match file_1: linear, short transition while animating; drop during reset to avoid back-glide
              d && !c && "transition-all duration-[50ms] ease-linear",
              !d && "transition-all duration-300 ease-out",
              w[s]
            ),
            style: { width: `${M}%` }
          }
        ),
        /* @__PURE__ */ p("div", { className: "content-stretch flex justify-between items-center relative flex-1 z-10", children: [
          /* @__PURE__ */ t("div", { className: "content-stretch flex gap-[3.386px] items-center relative shrink-0", children: /* @__PURE__ */ t("div", { className: "flex flex-col font-sans font-semibold justify-center relative shrink-0 text-video-title text-white tracking-[0.0466px] whitespace-nowrap", children: /* @__PURE__ */ t("p", { children: e }) }) }),
          o && /* @__PURE__ */ t("div", { className: "content-stretch flex gap-[3.386px] items-center relative shrink-0", children: /* @__PURE__ */ t(
            "button",
            {
              onClick: n,
              className: "bg-toast-cancel hover:bg-toast-cancel-hover active:bg-toast-cancel-active box-border content-stretch flex h-[20.317px] items-center px-[6.772px] py-[3.386px] relative rounded-[4.13px] shrink-0 transition-colors focus:outline-none focus:ring-2 focus:ring-white/20",
              children: /* @__PURE__ */ t("div", { className: "flex flex-col font-sans font-book justify-center not-italic relative shrink-0 text-video-title text-white tracking-[0.0466px] whitespace-nowrap", children: /* @__PURE__ */ t("p", { children: "Cancel" }) })
            }
          ) })
        ] })
      ]
    }
  );
}
const le = {
  p3349e200: "M10.25 15.25L1.25 8.25001L10.25 1.25001V15.25Z",
  p3708d380: "M1.25 15.25L10.25 8.25001L1.25 1.25001V15.25Z",
  p9253180: "M21.25 15.25L12.25 8.25001L21.25 1.25001V15.25Z",
  pdcdee80: "M12.25 15.25L21.25 8.25001L12.25 1.25001V15.25Z"
};
function It({
  currentTime: e = 0,
  totalTime: r = 0,
  progress: o = 0,
  isPlaying: n = !1,
  onPlayPause: s,
  onSeek: a,
  onSettings: i,
  onBackward: l,
  onForward: c,
  className: u
}) {
  const [d, b] = R(!1), [w, v] = R(e), M = X(null), C = (N) => {
    const I = Math.floor(N / 60), V = Math.floor(N % 60);
    return `${I}:${V.toString().padStart(2, "0")}`;
  }, k = K((N) => {
    if (!M.current || !a) return;
    const I = M.current.getBoundingClientRect(), G = (N.clientX - I.left) / I.width * 100;
    a(Math.max(0, Math.min(100, G)));
  }, [a]), A = K((N) => {
    b(!0), k(N);
  }, [k]), j = K((N) => {
    if (!d || !M.current || !a) return;
    const I = M.current.getBoundingClientRect(), G = (N.clientX - I.left) / I.width * 100;
    a(Math.max(0, Math.min(100, G)));
  }, [d, a]), T = K(() => {
    b(!1);
  }, []);
  return F(() => {
    if (d)
      return document.addEventListener("mousemove", j), document.addEventListener("mouseup", T), () => {
        document.removeEventListener("mousemove", j), document.removeEventListener("mouseup", T);
      };
  }, [d, j, T]), F(() => {
    if (n && w < r) {
      const N = setInterval(() => {
        v((I) => Math.min(r, I + 1));
      }, 1e3);
      return () => clearInterval(N);
    }
  }, [n, w, r]), F(() => {
    if (n && a && o < 100) {
      const N = setInterval(() => {
        a(Math.min(100, o + 0.5));
      }, 1e3);
      return () => clearInterval(N);
    }
  }, [n, a, o]), F(() => {
    v(e);
  }, [e]), /* @__PURE__ */ p("div", { className: h(
    "box-border flex flex-col gap-[12px] items-start p-[0px] w-full",
    u
  ), children: [
    /* @__PURE__ */ p(
      "div",
      {
        ref: M,
        className: "bg-bg-subtle h-[8px] relative rounded-full shrink-0 w-full cursor-pointer",
        onClick: k,
        children: [
          /* @__PURE__ */ t(
            "div",
            {
              className: "absolute bg-brand-primary h-[8px] left-0 rounded-full top-0 transition-all duration-150 ease-out",
              style: { width: `${o}%` }
            }
          ),
          /* @__PURE__ */ p(
            "div",
            {
              className: "absolute w-[50px] h-[50px] top-[-21px] cursor-grab active:cursor-grabbing transition-left duration-150 ease-out pointer-events-auto",
              style: { left: `calc(${o}% - 21px)` },
              onMouseDown: A,
              children: [
                d && /* @__PURE__ */ t("div", { className: "absolute left-1/2 -translate-x-1/2 bottom-[calc(100%+8px)] bg-surface-darkest text-white px-2 py-1 rounded text-xs whitespace-nowrap animate-in fade-in duration-150 select-none pointer-events-none", children: C(o / 100 * r) }),
                /* @__PURE__ */ p("svg", { className: "block w-full h-full", fill: "none", viewBox: "0 0 50 50", preserveAspectRatio: "xMidYMid meet", children: [
                  /* @__PURE__ */ p("g", { filter: "url(#filter0_d_0_13)", children: [
                    /* @__PURE__ */ t("circle", { cx: "25", cy: "25", fill: "white", r: "7" }),
                    /* @__PURE__ */ t("circle", { cx: "25", cy: "25", r: "6.5", stroke: "var(--border-primary)", strokeWidth: "1" })
                  ] }),
                  /* @__PURE__ */ t("defs", { children: /* @__PURE__ */ p("filter", { colorInterpolationFilters: "sRGB", filterUnits: "userSpaceOnUse", height: "48.2", id: "filter0_d_0_13", width: "48.2", x: "0.9", y: "0.9", children: [
                    /* @__PURE__ */ t("feFlood", { floodOpacity: "0", result: "BackgroundImageFix" }),
                    /* @__PURE__ */ t("feColorMatrix", { in: "SourceAlpha", result: "hardAlpha", type: "matrix", values: "0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0" }),
                    /* @__PURE__ */ t("feMorphology", { in: "SourceAlpha", operator: "dilate", radius: "5", result: "effect1_dropShadow_0_13" }),
                    /* @__PURE__ */ t("feOffset", {}),
                    /* @__PURE__ */ t("feGaussianBlur", { stdDeviation: "6.05" }),
                    /* @__PURE__ */ t("feComposite", { in2: "hardAlpha", operator: "out" }),
                    /* @__PURE__ */ t("feColorMatrix", { type: "matrix", values: "0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.1 0" }),
                    /* @__PURE__ */ t("feBlend", { in2: "BackgroundImageFix", mode: "normal", result: "effect1_dropShadow_0_13" }),
                    /* @__PURE__ */ t("feBlend", { in: "SourceGraphic", in2: "effect1_dropShadow_0_13", mode: "normal", result: "shape" })
                  ] }) })
                ] })
              ]
            }
          )
        ]
      }
    ),
    /* @__PURE__ */ p("div", { className: "flex items-center justify-between shrink-0 w-full h-[24px]", children: [
      /* @__PURE__ */ p("div", { className: "flex gap-[8px] items-center shrink-0", children: [
        /* @__PURE__ */ t(
          "button",
          {
            className: "overflow-clip relative shrink-0 size-[24px] hover:opacity-80 transition-opacity",
            onClick: s,
            children: /* @__PURE__ */ t("div", { className: "absolute inset-[12.5%_20.83%]", children: /* @__PURE__ */ t("svg", { className: "block size-full", fill: "none", preserveAspectRatio: "none", viewBox: "0 0 18 22", children: n ? /* @__PURE__ */ p("g", { children: [
              /* @__PURE__ */ t("rect", { x: "2", y: "2", width: "5", height: "20", fill: "var(--text-primary)", rx: "1" }),
              /* @__PURE__ */ t("rect", { x: "11", y: "2", width: "5", height: "20", fill: "var(--text-primary)", rx: "1" })
            ] }) : /* @__PURE__ */ t("path", { d: "M2 2L16 11L2 20V2Z", stroke: "var(--text-primary)", strokeLinecap: "round", strokeLinejoin: "round", strokeWidth: "2.5" }) }) })
          }
        ),
        /* @__PURE__ */ t(
          "button",
          {
            className: "overflow-clip relative shrink-0 size-[24px] hover:opacity-80 transition-opacity",
            onClick: l,
            children: /* @__PURE__ */ t("div", { className: "absolute inset-[20.83%_8.33%]", children: /* @__PURE__ */ t("div", { className: "absolute inset-[-8.93%_-6.25%]", style: { "--stroke-0": "var(--text-primary)" }, children: /* @__PURE__ */ p("svg", { className: "block size-full", fill: "none", preserveAspectRatio: "none", viewBox: "0 0 23 17", children: [
              /* @__PURE__ */ t("path", { d: le.p3349e200, stroke: "var(--stroke-0, black)", strokeLinecap: "round", strokeLinejoin: "round", strokeOpacity: "0.5", strokeWidth: "2" }),
              /* @__PURE__ */ t("path", { d: le.p9253180, stroke: "var(--stroke-0, black)", strokeLinecap: "round", strokeLinejoin: "round", strokeOpacity: "0.5", strokeWidth: "2" })
            ] }) }) })
          }
        ),
        /* @__PURE__ */ t(
          "button",
          {
            className: "overflow-clip relative shrink-0 size-[24px] hover:opacity-80 transition-opacity",
            onClick: c,
            children: /* @__PURE__ */ t("div", { className: "absolute inset-[20.83%_8.33%]", children: /* @__PURE__ */ t("div", { className: "absolute inset-[-8.93%_-6.25%]", style: { "--stroke-0": "var(--text-primary)" }, children: /* @__PURE__ */ p("svg", { className: "block size-full", fill: "none", preserveAspectRatio: "none", viewBox: "0 0 23 17", children: [
              /* @__PURE__ */ t("path", { d: le.pdcdee80, stroke: "var(--stroke-0, black)", strokeLinecap: "round", strokeLinejoin: "round", strokeOpacity: "0.5", strokeWidth: "2" }),
              /* @__PURE__ */ t("path", { d: le.p3708d380, stroke: "var(--stroke-0, black)", strokeLinecap: "round", strokeLinejoin: "round", strokeOpacity: "0.5", strokeWidth: "2" })
            ] }) }) })
          }
        )
      ] }),
      /* @__PURE__ */ p("div", { className: "flex gap-[3px] items-center", children: [
        /* @__PURE__ */ t("p", { className: "font-medium leading-[20px] shrink-0 text-[14px] text-text-secondary", children: C(w) }),
        /* @__PURE__ */ t("div", { className: "h-[10px] shrink-0 w-[6px]", children: /* @__PURE__ */ t("svg", { className: "block size-full", fill: "none", preserveAspectRatio: "none", viewBox: "0 0 8 12", children: /* @__PURE__ */ t("path", { d: "M1 11L7 1", stroke: "var(--text-secondary)", strokeWidth: "1.5" }) }) }),
        /* @__PURE__ */ t("p", { className: "font-medium leading-[20px] shrink-0 text-[14px] text-text-secondary", children: C(r) })
      ] }),
      /* @__PURE__ */ t("div", { className: "flex gap-[4px] h-[24px] items-center shrink-0", children: /* @__PURE__ */ t(
        "button",
        {
          className: "overflow-clip relative shrink-0 size-[24px] hover:opacity-80 transition-opacity",
          onClick: i,
          children: /* @__PURE__ */ t("div", { className: "absolute inset-[4.167%]", children: /* @__PURE__ */ p("svg", { className: "block size-full", fill: "none", preserveAspectRatio: "none", viewBox: "0 0 26 26", children: [
            /* @__PURE__ */ t("path", { d: "M13 16C14.6569 16 16 14.6569 16 13C16 11.3431 14.6569 10 13 10C11.3431 10 10 11.3431 10 13C10 14.6569 11.3431 16 13 16Z", stroke: "var(--text-secondary)", strokeLinecap: "round", strokeLinejoin: "round", strokeWidth: "2.5" }),
            /* @__PURE__ */ t("path", { d: "M20.4 16C20.2669 16.3016 20.2272 16.6362 20.286 16.9606C20.3448 17.285 20.4995 17.5843 20.73 17.82L20.79 17.88C20.976 18.0657 21.1235 18.2863 21.2241 18.5291C21.3248 18.7719 21.3766 19.0322 21.3766 19.295C21.3766 19.5578 21.3248 19.8181 21.2241 20.0609C21.1235 20.3037 20.976 20.5243 20.79 20.71C20.6043 20.896 20.3837 21.0435 20.1409 21.1441C19.8981 21.2448 19.6378 21.2966 19.375 21.2966C19.1122 21.2966 18.8519 21.2448 18.6091 21.1441C18.3663 21.0435 18.1457 20.896 17.96 20.71L17.9 20.65C17.6643 20.4195 17.365 20.2648 17.0406 20.206C16.7162 20.1472 16.3816 20.1869 16.08 20.32C15.7842 20.4468 15.532 20.6572 15.3543 20.9255C15.1766 21.1938 15.0813 21.5082 15.08 21.83V22C15.08 22.5304 14.8693 23.0391 14.4942 23.4142C14.1191 23.7893 13.6104 24 13.08 24C12.5496 24 12.0409 23.7893 11.6658 23.4142C11.2907 23.0391 11.08 22.5304 11.08 22V21.91C11.0723 21.579 10.9651 21.258 10.7725 20.9887C10.5799 20.7194 10.3107 20.5143 10 20.4C9.69838 20.2669 9.36381 20.2272 9.03941 20.286C8.71502 20.3448 8.41568 20.4995 8.18 20.73L8.12 20.79C7.93425 20.976 7.71368 21.1235 7.47088 21.2241C7.22808 21.3248 6.96783 21.3766 6.705 21.3766C6.44217 21.3766 6.18192 21.3248 5.93912 21.2241C5.69632 21.1235 5.47575 20.976 5.29 20.79C5.10405 20.6043 4.95653 20.3837 4.85588 20.1409C4.75523 19.8981 4.70343 19.6378 4.70343 19.375C4.70343 19.1122 4.75523 18.8519 4.85588 18.6091C4.95653 18.3663 5.10405 18.1457 5.29 17.96L5.35 17.9C5.58054 17.6643 5.73519 17.365 5.794 17.0406C5.85282 16.7162 5.81312 16.3816 5.68 16.08C5.55324 15.7842 5.34276 15.532 5.07447 15.3543C4.80618 15.1766 4.49179 15.0813 4.17 15.08H4C3.46957 15.08 2.96086 14.8693 2.58579 14.4942C2.21071 14.1191 2 13.6104 2 13.08C2 12.5496 2.21071 12.0409 2.58579 11.6658C2.96086 11.2907 3.46957 11.08 4 11.08H4.09C4.42099 11.0723 4.742 10.9651 5.0113 10.7725C5.28059 10.5799 5.48572 10.3107 5.6 10C5.73312 9.69838 5.77282 9.36381 5.714 9.03941C5.65519 8.71502 5.50054 8.41568 5.27 8.18L5.21 8.12C5.02405 7.93425 4.87653 7.71368 4.77588 7.47088C4.67523 7.22808 4.62343 6.96783 4.62343 6.705C4.62343 6.44217 4.67523 6.18192 4.77588 5.93912C4.87653 5.69632 5.02405 5.47575 5.21 5.29C5.39575 5.10405 5.61632 4.95653 5.85912 4.85588C6.10192 4.75523 6.36217 4.70343 6.625 4.70343C6.88783 4.70343 7.14808 4.75523 7.39088 4.85588C7.63368 4.95653 7.85425 5.10405 8.04 5.29L8.1 5.35C8.33568 5.58054 8.63502 5.73519 8.95941 5.794C9.28381 5.85282 9.61838 5.81312 9.92 5.68H10C10.2958 5.55324 10.548 5.34276 10.7257 5.07447C10.9034 4.80618 10.9987 4.49179 11 4.17V4C11 3.46957 11.2107 2.96086 11.5858 2.58579C11.9609 2.21071 12.4696 2 13 2C13.5304 2 14.0391 2.21071 14.4142 2.58579C14.7893 2.96086 15 3.46957 15 4V4.09C15.0013 4.41179 15.0966 4.72618 15.2743 4.99447C15.452 5.26276 15.7042 5.47324 16 5.6C16.3016 5.73312 16.6362 5.77282 16.9606 5.714C17.285 5.65519 17.5843 5.50054 17.82 5.27L17.88 5.21C18.0657 5.02405 18.2863 4.87653 18.5291 4.77588C18.7719 4.67523 19.0322 4.62343 19.295 4.62343C19.5578 4.62343 19.8181 4.67523 20.0609 4.77588C20.3037 4.87653 20.5243 5.02405 20.71 5.21C20.896 5.39575 21.0435 5.61632 21.1441 5.85912C21.2448 6.10192 21.2966 6.36217 21.2966 6.625C21.2966 6.88783 21.2448 7.14808 21.1441 7.39088C21.0435 7.63368 20.896 7.85425 20.71 8.04L20.65 8.1C20.4195 8.33568 20.2648 8.63502 20.206 8.95941C20.1472 9.28381 20.1869 9.61838 20.32 9.92V10C20.4468 10.2958 20.6572 10.548 20.9255 10.7257C21.1938 10.9034 21.5082 10.9987 21.83 11H22C22.5304 11 23.0391 11.2107 23.4142 11.5858C23.7893 11.9609 24 12.4696 24 13C24 13.5304 23.7893 14.0391 23.4142 14.4142C23.0391 14.7893 22.5304 15 22 15H21.91C21.5882 15.0013 21.2738 15.0966 21.0055 15.2743C20.7372 15.452 20.5268 15.7042 20.4 16Z", stroke: "var(--text-secondary)", strokeLinecap: "round", strokeLinejoin: "round", strokeWidth: "2.5" })
          ] }) })
        }
      ) })
    ] })
  ] });
}
function St({
  label: e,
  description: r,
  defaultChecked: o = !1,
  onChange: n,
  disabled: s = !1,
  className: a,
  ...i
}) {
  const [l, c] = R(o), u = () => {
    if (s) return;
    const d = !l;
    c(d), n == null || n(d);
  };
  return /* @__PURE__ */ p(
    "label",
    {
      className: h(
        "flex gap-3 items-start cursor-pointer",
        s && "opacity-50 cursor-not-allowed",
        a
      ),
      children: [
        /* @__PURE__ */ t(
          "button",
          {
            role: "checkbox",
            type: "button",
            "aria-checked": l,
            onClick: u,
            disabled: s,
            className: h(
              "relative flex items-center justify-center size-5 shrink-0 rounded mt-0.5",
              "border transition-all duration-200",
              "focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-brand-primary",
              "disabled:cursor-not-allowed",
              l ? "bg-brand-primary border-brand-primary" : "bg-transparent border-border-primary hover:border-text-secondary"
            ),
            children: l && /* @__PURE__ */ t(
              "svg",
              {
                width: "12",
                height: "12",
                viewBox: "0 0 12 12",
                fill: "none",
                className: "text-on-brand",
                children: /* @__PURE__ */ t(
                  "path",
                  {
                    d: "M2.5 6L5 8.5L9.5 3.5",
                    stroke: "currentColor",
                    strokeWidth: "1.75",
                    strokeLinecap: "round",
                    strokeLinejoin: "round"
                  }
                )
              }
            )
          }
        ),
        (e || r) && /* @__PURE__ */ p("div", { className: "flex flex-col gap-1 min-w-0", children: [
          e && /* @__PURE__ */ t("span", { className: "text-label text-text-primary", children: e }),
          r && /* @__PURE__ */ t("span", { className: "text-label-sm text-text-secondary", children: r })
        ] })
      ]
    }
  );
}
function Rt({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ t(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: /* @__PURE__ */ t("polygon", { points: "5 3 19 12 5 21 5 3" })
    }
  );
}
function _t({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ p(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: [
        /* @__PURE__ */ t("path", { d: "m3 9 9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z" }),
        /* @__PURE__ */ t("polyline", { points: "9 22 9 12 15 12 15 22" })
      ]
    }
  );
}
function Bt({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ p(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: [
        /* @__PURE__ */ t("rect", { width: "18", height: "18", x: "3", y: "3", rx: "2" }),
        /* @__PURE__ */ t("path", { d: "M7 3v18" }),
        /* @__PURE__ */ t("path", { d: "M3 7.5h4" }),
        /* @__PURE__ */ t("path", { d: "M3 12h18" }),
        /* @__PURE__ */ t("path", { d: "M3 16.5h4" }),
        /* @__PURE__ */ t("path", { d: "M17 3v18" }),
        /* @__PURE__ */ t("path", { d: "M17 7.5h4" }),
        /* @__PURE__ */ t("path", { d: "M17 16.5h4" })
      ]
    }
  );
}
function Tt({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ t(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: /* @__PURE__ */ t("path", { d: "M4 19.5v-15A2.5 2.5 0 0 1 6.5 2H20v20H6.5a2.5 2.5 0 0 1 0-5H20" })
    }
  );
}
function Vt({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ t(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: /* @__PURE__ */ t("path", { d: "M20 20a2 2 0 0 0 2-2V8a2 2 0 0 0-2-2h-7.9a2 2 0 0 1-1.69-.9L9.6 3.9A2 2 0 0 0 7.93 3H4a2 2 0 0 0-2 2v13a2 2 0 0 0 2 2Z" })
    }
  );
}
function Pt({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ p(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: [
        /* @__PURE__ */ t("path", { d: "M12.22 2h-.44a2 2 0 0 0-2 2v.18a2 2 0 0 1-1 1.73l-.43.25a2 2 0 0 1-2 0l-.15-.08a2 2 0 0 0-2.73.73l-.22.38a2 2 0 0 0 .73 2.73l.15.1a2 2 0 0 1 1 1.72v.51a2 2 0 0 1-1 1.74l-.15.09a2 2 0 0 0-.73 2.73l.22.38a2 2 0 0 0 2.73.73l.15-.08a2 2 0 0 1 2 0l.43.25a2 2 0 0 1 1 1.73V20a2 2 0 0 0 2 2h.44a2 2 0 0 0 2-2v-.18a2 2 0 0 1 1-1.73l.43-.25a2 2 0 0 1 2 0l.15.08a2 2 0 0 0 2.73-.73l.22-.39a2 2 0 0 0-.73-2.73l-.15-.08a2 2 0 0 1-1-1.74v-.5a2 2 0 0 1 1-1.74l.15-.09a2 2 0 0 0 .73-2.73l-.22-.38a2 2 0 0 0-2.73-.73l-.15.08a2 2 0 0 1-2 0l-.43-.25a2 2 0 0 1-1-1.73V4a2 2 0 0 0-2-2z" }),
        /* @__PURE__ */ t("circle", { cx: "12", cy: "12", r: "3" })
      ]
    }
  );
}
function Et({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ p(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: [
        /* @__PURE__ */ t("path", { d: "m16 13 5.223 3.482a.5.5 0 0 0 .777-.416V7.87a.5.5 0 0 0-.752-.432L16 10.5" }),
        /* @__PURE__ */ t("rect", { x: "2", y: "6", width: "14", height: "12", rx: "2" })
      ]
    }
  );
}
function Gt({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ p(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: [
        /* @__PURE__ */ t("circle", { cx: "11", cy: "11", r: "8" }),
        /* @__PURE__ */ t("path", { d: "m21 21-4.3-4.3" })
      ]
    }
  );
}
function Ot({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ p(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: [
        /* @__PURE__ */ t("circle", { cx: "6", cy: "6", r: "3" }),
        /* @__PURE__ */ t("path", { d: "M8.12 8.12 12 12" }),
        /* @__PURE__ */ t("path", { d: "M20 4 8.12 15.88" }),
        /* @__PURE__ */ t("circle", { cx: "6", cy: "18", r: "3" }),
        /* @__PURE__ */ t("path", { d: "M14.8 14.8 20 20" })
      ]
    }
  );
}
function Ft({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ p(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: [
        /* @__PURE__ */ t("line", { x1: "4", x2: "4", y1: "21", y2: "14" }),
        /* @__PURE__ */ t("line", { x1: "4", x2: "4", y1: "10", y2: "3" }),
        /* @__PURE__ */ t("line", { x1: "12", x2: "12", y1: "21", y2: "12" }),
        /* @__PURE__ */ t("line", { x1: "12", x2: "12", y1: "8", y2: "3" }),
        /* @__PURE__ */ t("line", { x1: "20", x2: "20", y1: "21", y2: "16" }),
        /* @__PURE__ */ t("line", { x1: "20", x2: "20", y1: "12", y2: "3" }),
        /* @__PURE__ */ t("line", { x1: "2", x2: "6", y1: "14", y2: "14" }),
        /* @__PURE__ */ t("line", { x1: "10", x2: "14", y1: "8", y2: "8" }),
        /* @__PURE__ */ t("line", { x1: "18", x2: "22", y1: "16", y2: "16" })
      ]
    }
  );
}
function Wt({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ p(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: [
        /* @__PURE__ */ t("polyline", { points: "4 7 4 4 20 4 20 7" }),
        /* @__PURE__ */ t("line", { x1: "9", x2: "15", y1: "20", y2: "20" }),
        /* @__PURE__ */ t("line", { x1: "12", x2: "12", y1: "4", y2: "20" })
      ]
    }
  );
}
function Dt({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ p(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: [
        /* @__PURE__ */ t("path", { d: "m21.64 3.64-1.28-1.28a1.21 1.21 0 0 0-1.72 0L2.36 18.64a1.21 1.21 0 0 0 0 1.72l1.28 1.28a1.2 1.2 0 0 0 1.72 0L21.64 5.36a1.2 1.2 0 0 0 0-1.72Z" }),
        /* @__PURE__ */ t("path", { d: "m14 7 3 3" }),
        /* @__PURE__ */ t("path", { d: "M5 6v4" }),
        /* @__PURE__ */ t("path", { d: "M19 14v4" }),
        /* @__PURE__ */ t("path", { d: "M10 2v2" }),
        /* @__PURE__ */ t("path", { d: "M7 8H3" }),
        /* @__PURE__ */ t("path", { d: "M21 16h-4" }),
        /* @__PURE__ */ t("path", { d: "M11 3H9" })
      ]
    }
  );
}
function $t({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ p(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: [
        /* @__PURE__ */ t("path", { d: "M12 2a3 3 0 0 0-3 3v7a3 3 0 0 0 6 0V5a3 3 0 0 0-3-3Z" }),
        /* @__PURE__ */ t("path", { d: "M19 10v2a7 7 0 0 1-14 0v-2" }),
        /* @__PURE__ */ t("line", { x1: "12", x2: "12", y1: "19", y2: "22" })
      ]
    }
  );
}
function Ht({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ p(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: [
        /* @__PURE__ */ t("path", { d: "M9 18V5l12-2v13" }),
        /* @__PURE__ */ t("circle", { cx: "6", cy: "18", r: "3" }),
        /* @__PURE__ */ t("circle", { cx: "18", cy: "16", r: "3" })
      ]
    }
  );
}
function Ut({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ p(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: [
        /* @__PURE__ */ t("path", { d: "M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4" }),
        /* @__PURE__ */ t("polyline", { points: "17 8 12 3 7 8" }),
        /* @__PURE__ */ t("line", { x1: "12", x2: "12", y1: "3", y2: "15" })
      ]
    }
  );
}
function ct({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ p(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: [
        /* @__PURE__ */ t("path", { d: "M18 6 6 18" }),
        /* @__PURE__ */ t("path", { d: "m6 6 12 12" })
      ]
    }
  );
}
function Zt({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ p(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: [
        /* @__PURE__ */ t("path", { d: "m3 3 7.07 16.97 2.51-7.39 7.39-2.51L3 3z" }),
        /* @__PURE__ */ t("path", { d: "m13 13 6 6" })
      ]
    }
  );
}
function Yt({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ p(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: [
        /* @__PURE__ */ t("path", { d: "M6 2v14a2 2 0 0 0 2 2h14" }),
        /* @__PURE__ */ t("path", { d: "M18 22V8a2 2 0 0 0-2-2H2" })
      ]
    }
  );
}
function qt({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s }) {
  return /* @__PURE__ */ t(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: "none",
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: /* @__PURE__ */ t("path", { d: "M7.9 20A9 9 0 1 0 4 16.1L2 22z" })
    }
  );
}
function dt({ className: e, size: r = 24, strokeWidth: o = 2, opacity: n, color: s, fill: a = "none" }) {
  return /* @__PURE__ */ t(
    "svg",
    {
      width: r,
      height: r,
      viewBox: "0 0 24 24",
      fill: a,
      stroke: s || "currentColor",
      strokeWidth: o,
      strokeLinecap: "round",
      strokeLinejoin: "round",
      className: e,
      opacity: n,
      children: /* @__PURE__ */ t("polygon", { points: "12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2" })
    }
  );
}
function Kt({
  defaultFavorited: e = !1,
  onChange: r,
  className: o,
  ...n
}) {
  const [s, a] = R(e);
  return /* @__PURE__ */ t(
    "button",
    {
      onClick: () => {
        const l = !s;
        a(l), r == null || r(l);
      },
      className: h(
        "flex items-center justify-center size-6 rounded-full bg-surface-bg overflow-clip shrink-0",
        "transition-colors focus:outline-none focus-visible:ring-2 focus-visible:ring-brand-primary",
        o
      ),
      "aria-label": s ? "Remove from favorites" : "Add to favorites",
      ...n,
      children: /* @__PURE__ */ t(
        dt,
        {
          size: 14,
          strokeWidth: 2,
          color: s ? "var(--brand-primary)" : "currentColor",
          fill: s ? "var(--brand-primary)" : "none"
        }
      )
    }
  );
}
function Xt({
  isOpen: e,
  onClose: r,
  title: o,
  children: n,
  footer: s,
  size: a = "medium",
  className: i
}) {
  const l = K((u) => {
    u.key === "Escape" && r();
  }, [r]);
  return F(() => (e && (document.addEventListener("keydown", l), document.body.style.overflow = "hidden"), () => {
    document.removeEventListener("keydown", l), document.body.style.overflow = "";
  }), [e, l]), e ? /* @__PURE__ */ p(rr, { children: [
    /* @__PURE__ */ t(
      "div",
      {
        className: "fixed inset-0 bg-modal-scrim z-50 transition-opacity",
        onClick: r
      }
    ),
    /* @__PURE__ */ p(
      "div",
      {
        role: "dialog",
        "aria-modal": "true",
        "aria-label": o,
        className: h(
          "fixed left-1/2 top-1/2 -translate-x-1/2 -translate-y-1/2 z-50",
          "bg-surface-bg rounded-2xl shadow-xl max-h-[90vh] overflow-hidden flex flex-col",
          {
            small: "w-[360px]",
            medium: "w-[480px]",
            large: "w-[640px]"
          }[a],
          i
        ),
        children: [
          o && /* @__PURE__ */ p("div", { className: "flex items-center justify-between px-6 py-5 border-b border-border-secondary", children: [
            /* @__PURE__ */ t("h2", { className: "text-heading text-text-primary", children: o }),
            /* @__PURE__ */ t(
              "button",
              {
                onClick: r,
                className: "size-6 flex items-center justify-center rounded-md hover:bg-bg-hover transition-colors",
                children: /* @__PURE__ */ t(ct, { size: 16, color: "var(--text-secondary)" })
              }
            )
          ] }),
          /* @__PURE__ */ t("div", { className: "flex-1 overflow-y-auto px-6 py-6", children: n }),
          s && /* @__PURE__ */ t("div", { className: "flex items-center justify-end gap-3 px-6 py-5 border-t border-border-secondary", children: s })
        ]
      }
    )
  ] }) : null;
}
function Jt({
  options: e,
  value: r,
  defaultValue: o = "",
  onChange: n,
  name: s,
  disabled: a = !1,
  className: i
}) {
  const [l, c] = R(o), u = r !== void 0 ? r : l, d = (b) => {
    a || (r === void 0 && c(b), n == null || n(b));
  };
  return /* @__PURE__ */ t("div", { role: "radiogroup", className: h("flex flex-col gap-3", i), children: e.map((b) => {
    const w = u === b.value;
    return /* @__PURE__ */ p(
      "label",
      {
        className: h(
          "flex gap-3 items-start cursor-pointer",
          a && "opacity-50 cursor-not-allowed"
        ),
        children: [
          /* @__PURE__ */ t(
            "button",
            {
              role: "radio",
              type: "button",
              "aria-checked": w,
              onClick: () => d(b.value),
              disabled: a,
              className: h(
                "relative flex items-center justify-center size-5 shrink-0 rounded-full mt-0.5",
                "border transition-all duration-200",
                "focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-brand-primary",
                "disabled:cursor-not-allowed",
                w ? "border-brand-primary" : "border-border-primary hover:border-text-secondary"
              ),
              children: w && /* @__PURE__ */ t("div", { className: "size-2.5 rounded-full bg-brand-primary" })
            }
          ),
          (b.label || b.description) && /* @__PURE__ */ p("div", { className: "flex flex-col gap-1 min-w-0", children: [
            b.label && /* @__PURE__ */ t("span", { className: "text-label text-text-primary", children: b.label }),
            b.description && /* @__PURE__ */ t("span", { className: "text-label-sm text-text-secondary", children: b.description })
          ] })
        ]
      },
      b.value
    );
  }) });
}
function Qt({
  tabs: e,
  defaultTab: r,
  onChange: o,
  className: n
}) {
  var c, u;
  const [s, a] = R(r || ((c = e[0]) == null ? void 0 : c.id) || ""), i = (d) => {
    a(d), o == null || o(d);
  }, l = (u = e.find((d) => d.id === s)) == null ? void 0 : u.content;
  return /* @__PURE__ */ p("div", { className: h("flex flex-col", n), children: [
    /* @__PURE__ */ t("div", { role: "tablist", className: "flex border-b border-border-secondary", children: e.map((d) => {
      const b = s === d.id;
      return /* @__PURE__ */ p(
        "button",
        {
          role: "tab",
          "aria-selected": b,
          "aria-controls": `tabpanel-${d.id}`,
          onClick: () => i(d.id),
          className: h(
            "px-4 py-2.5 text-label-sm font-medium transition-colors relative",
            "focus:outline-none focus-visible:ring-2 focus-visible:ring-brand-primary focus-visible:ring-inset",
            b ? "text-brand-primary" : "text-text-secondary hover:text-text-primary"
          ),
          children: [
            d.label,
            b && /* @__PURE__ */ t("div", { className: "absolute bottom-0 left-0 right-0 h-0.5 bg-brand-primary" })
          ]
        },
        d.id
      );
    }) }),
    /* @__PURE__ */ t(
      "div",
      {
        role: "tabpanel",
        id: `tabpanel-${s}`,
        className: "pt-4",
        children: l
      }
    )
  ] });
}
function eo({
  children: e,
  className: r
}) {
  return /* @__PURE__ */ t(
    "div",
    {
      className: h(
        "inline-flex items-center justify-center gap-2 px-2",
        "h-12 rounded-2xl",
        "bg-surface-bg border border-border-primary",
        "shadow-[0px_0px_12.1px_0px_rgba(0,0,0,0.1)]",
        r
      ),
      children: e
    }
  );
}
function ro({
  icon: e,
  selected: r = !1,
  className: o,
  ...n
}) {
  return /* @__PURE__ */ t(
    "button",
    {
      className: h(
        "flex items-center justify-center size-8 rounded-lg overflow-clip shrink-0 transition-colors",
        "focus:outline-none focus-visible:ring-2 focus-visible:ring-brand-primary",
        r ? "bg-brand-secondary" : "hover:bg-bg-hover",
        o
      ),
      ...n,
      children: /* @__PURE__ */ t("span", { className: h(
        "flex items-center justify-center",
        r ? "opacity-85" : "opacity-50"
      ), children: e })
    }
  );
}
function to({
  content: e,
  position: r = "top",
  delay: o = 200,
  children: n,
  className: s
}) {
  const [a, i] = R(!1), l = X(null), c = K(() => {
    l.current = setTimeout(() => i(!0), o);
  }, [o]), u = K(() => {
    l.current && clearTimeout(l.current), i(!1);
  }, []);
  F(() => () => {
    l.current && clearTimeout(l.current);
  }, []);
  const d = {
    top: "bottom-full left-1/2 -translate-x-1/2 mb-2",
    bottom: "top-full left-1/2 -translate-x-1/2 mt-2",
    left: "right-full top-1/2 -translate-y-1/2 mr-2",
    right: "left-full top-1/2 -translate-y-1/2 ml-2"
  }, b = {
    top: "top-full left-1/2 -translate-x-1/2 border-t-surface-dark border-x-transparent border-b-transparent",
    bottom: "bottom-full left-1/2 -translate-x-1/2 border-b-surface-dark border-x-transparent border-t-transparent",
    left: "left-full top-1/2 -translate-y-1/2 border-l-surface-dark border-y-transparent border-r-transparent",
    right: "right-full top-1/2 -translate-y-1/2 border-r-surface-dark border-y-transparent border-l-transparent"
  };
  return /* @__PURE__ */ p(
    "div",
    {
      className: "relative inline-flex",
      onMouseEnter: c,
      onMouseLeave: u,
      onFocus: c,
      onBlur: u,
      children: [
        n,
        a && /* @__PURE__ */ t(
          "div",
          {
            role: "tooltip",
            className: h(
              "absolute z-50 pointer-events-none",
              d[r]
            ),
            children: /* @__PURE__ */ p(
              "div",
              {
                className: h(
                  "bg-surface-dark text-label-sm px-3 py-1.5 rounded-lg whitespace-nowrap",
                  "animate-in fade-in duration-150",
                  s
                ),
                style: { color: "var(--on-brand)" },
                children: [
                  e,
                  /* @__PURE__ */ t(
                    "div",
                    {
                      className: h(
                        "absolute w-0 h-0 border-4",
                        b[r]
                      )
                    }
                  )
                ]
              }
            )
          }
        )
      ]
    }
  );
}
function oo({
  icon: e,
  label: r,
  active: o = !1,
  className: n,
  ...s
}) {
  return /* @__PURE__ */ p(
    "button",
    {
      className: h(
        "flex items-center gap-lg px-lg py-lg w-full rounded-corner-md transition-colors cursor-pointer text-left",
        o ? "bg-brand-tertiary text-brand-primary" : "text-text-secondary hover:bg-bg-hover",
        n
      ),
      "aria-current": o ? "page" : void 0,
      ...s,
      children: [
        /* @__PURE__ */ t("span", { className: "size-5 shrink-0", children: e }),
        /* @__PURE__ */ t("span", { className: "text-label-sm whitespace-nowrap", children: r })
      ]
    }
  );
}
function no({
  title: e,
  children: r,
  className: o
}) {
  return /* @__PURE__ */ p(
    "nav",
    {
      className: h(
        "flex flex-col gap-xl bg-surface-secondary-bg border-r border-border-primary px-xl py-lg w-[252px] h-full",
        o
      ),
      children: [
        /* @__PURE__ */ t("div", { className: "py-lg", children: /* @__PURE__ */ t("h2", { className: "text-heading text-text-primary", children: e }) }),
        /* @__PURE__ */ t("div", { className: "flex flex-col gap-xs flex-1", children: r })
      ]
    }
  );
}
const Qe = tr(void 0), er = "astra-theme";
function ut() {
  if (typeof window > "u") return "light";
  const e = localStorage.getItem(er);
  return e === "dark" || e === "light" ? e : window.matchMedia("(prefers-color-scheme: dark)").matches ? "dark" : "light";
}
function so({ children: e }) {
  const [r, o] = R(ut);
  F(() => {
    const a = document.documentElement;
    r === "dark" ? a.classList.add("dark") : a.classList.remove("dark"), localStorage.setItem(er, r);
  }, [r]);
  const n = () => o((a) => a === "light" ? "dark" : "light"), s = (a) => o(a);
  return /* @__PURE__ */ t(Qe.Provider, { value: { theme: r, toggleTheme: n, setTheme: s }, children: e });
}
function ao() {
  const e = or(Qe);
  if (!e) throw new Error("useTheme must be used within ThemeProvider");
  return e;
}
export {
  ht as AIVideoCreation,
  Yr as AstraLogo,
  qr as Avatar,
  ft as AvatarGroup,
  qe as Badge,
  Tt as Book,
  bt as Button,
  xt as ButtonGroup,
  gt as ChatBubbles,
  St as Checkbox,
  Yt as Crop,
  lt as DurationBadge,
  Kt as FavoriteButton,
  Bt as Film,
  Vt as Folder,
  _t as Home,
  Kr as IconButton,
  kt as InputField,
  Ct as ItemCard,
  qt as MessageCircle,
  $t as Mic,
  Xt as Modal,
  Zt as MousePointer,
  Ht as Music,
  Rt as Play,
  Je as PromptInput,
  vt as PromptPane,
  Jt as RadioGroup,
  Ot as Scissors,
  Gt as Search,
  yt as SearchComponent,
  no as SecondaryNav,
  oo as SecondaryNavItem,
  wt as SegmentedControl,
  Nt as SelectField,
  Pt as Settings,
  Lt as SidebarButton,
  zt as SidebarNavigation,
  Ft as Sliders,
  dt as Star,
  Mt as SwitchField,
  Qt as Tabs,
  At as TextareaField,
  so as ThemeProvider,
  jt as Toast,
  eo as Toolbar,
  ro as ToolbarItem,
  to as Tooltip,
  Wt as Type,
  Ut as Upload,
  Et as Video,
  It as VideoControl,
  Dt as Wand2,
  ct as X,
  h as cn,
  ao as useTheme
};
